<!DOCTYPE html>
<!--[if lte IE 8 ]><html lang="en" class="no-js oldIE"><![endif]-->
<!--[if (gt IE 8)|!(IE)]><!--><html lang="en" class="no-js"><!--<![endif]-->
<head>
    <title>Adaptive efficient compression of genomes | Algorithms for Molecular Biology | Full Text</title>
    
    
    <meta name="journal_id" content='13015'/>
    
    <meta name="dc.identifier" content='10.1186/1748-7188-7-30'/>
    
    <meta name="DOI" content='10.1186/1748-7188-7-30'/>
    
    <meta name="dc.title" content='Adaptive efficient compression of genomes'/>
    
    <meta name="dc.description" content='Modern high-throughput sequencing technologies are able to generate DNA sequences at an ever increasing rate. In parallel to the decreasing experimental time and cost necessary to produce DNA sequences, computational requirements for analysis and storage of the sequences are steeply increasing. Compression is a key technology to deal with this challenge. Recently, referential compression schemes, storing only the differences between a to-be-compressed input and a known reference sequence, gained a lot of interest in this field. However, memory requirements of the current algorithms are high and run times often are slow. In this paper, we propose an adaptive, parallel and highly efficient referential sequence compression method which allows fine-tuning of the trade-off between required memory and compression speed. When using 12 MB of memory, our method is for human genomes on-par with the best previous algorithms in terms of compression ratio (400:1) and compression speed. In contrast, it compresses a complete human genome in just 11 seconds when provided with 9 GB of main memory, which is almost three times faster than the best competitor while using less main memory.'/>
    
    <meta name="dc.source" content='Algorithms for Molecular Biology 2012 7:1'/>
    
    <meta name="dc.format" content='text/html'/>
    
    <meta name="dc.publisher" content='BioMed Central'/>
    
    <meta name="dc.date" content='2012-11-12'/>
    
    <meta name="dc.type" content='OriginalPaper'/>
    
    <meta name="dc.language" content='En'/>
    
    <meta name="dc.copyright" content='2012 Wandelt and Leser; licensee BioMed Central Ltd.'/>
    
    <meta name="dc.rights" content='http://creativecommons.org/licenses/by/2.0'/>
    
    <meta name="dc.rightsAgent" content='reprints@biomedcentral.com'/>
    
    <meta name="prism.issn" content='1748-7188'/>
    
    <meta name="prism.doi" content='10.1186/1748-7188-7-30'/>
    
    <meta name="prism.publicationName" content='Algorithms for Molecular Biology'/>
    
    <meta name="prism.publicationDate" content='2012-11-12'/>
    
    <meta name="prism.volume" content='7'/>
    
    <meta name="prism.number" content='1'/>
    
    <meta name="prism.section" content='OriginalPaper'/>
    
    <meta name="prism.startingPage" content='30'/>
    
    <meta name="prism.copyright" content='2012 Wandelt and Leser; licensee BioMed Central Ltd.'/>
    
    <meta name="prism.rightsAgent" content='reprints@biomedcentral.com'/>
    
    <meta name="prism.url" content='http://almob.biomedcentral.com/articles/10.1186/1748-7188-7-30'/>
    
    <meta name="citation_pdf_url" content='http://almob.biomedcentral.com/track/pdf/10.1186/1748-7188-7-30?site=almob.biomedcentral.com'/>
    
    <meta name="citation_abstract_html_url" content='http://almob.biomedcentral.com/articles/10.1186/1748-7188-7-30'/>
    
    <meta name="citation_fulltext_html_url" content='http://almob.biomedcentral.com/articles/10.1186/1748-7188-7-30'/>
    
    <meta name="citation_journal_title" content='Algorithms for Molecular Biology'/>
    
    <meta name="citation_publisher" content='BioMed Central'/>
    
    <meta name="citation_issn" content='1748-7188'/>
    
    <meta name="citation_title" content='Adaptive efficient compression of genomes'/>
    
    <meta name="citation_volume" content='7'/>
    
    <meta name="citation_issue" content='1'/>
    
    <meta name="citation_date" content='2012/11/12'/>
    
    <meta name="citation_publication_date" content='2012/11/12'/>
    
    <meta name="citation_firstpage" content='30'/>
    
    <meta name="citation_doi" content='10.1186/1748-7188-7-30'/>
    
    <meta name="citation_fulltext_world_readable" content=''/>
    
    <meta name="dc.creator" content='Sebastian Wandelt'/>
    
    <meta name="dc.creator" content='Ulf Leser'/>
    
    <meta name="dc.subject" content='Bioinformatics'/>
    
    <meta name="dc.subject" content='Computational Biology/Bioinformatics'/>
    
    <meta name="dc.subject" content='Physiological, Cellular and Medical Topics'/>
    
    <meta name="dc.subject" content='Algorithms'/>
    
    <meta name="citation_author" content='Sebastian Wandelt'/>
    
    <meta name="citation_author_institution" content='Institute for Computer Science, Humboldt-Universit&#228;t zu Berlin, Berlin, Germany'/>
    
    <meta name="citation_author" content='Ulf Leser'/>
    
    <meta name="citation_author_institution" content='Institute for Computer Science, Humboldt-Universit&#228;t zu Berlin, Berlin, Germany'/>
    
    <meta name="citation_reference" content='citation_journal_title=Nature; citation_title=Initial sequencing and analysis of the human genome; citation_volume=409; citation_issue=6822; citation_publication_date=2001; citation_pages=860-921; citation_doi=10.1038/35057062; citation_id=CR1;'/>
    
    <meta name="citation_reference" content='citation_journal_title=Human Mol Genet; citation_title=A window into third-generation sequencing; citation_author=EE Schadt; citation_author=S Turner; citation_author=A Kasarskis; citation_volume=19; citation_issue=R2; citation_publication_date=2010; citation_pages=R227-R240; citation_doi=10.1093/hmg/ddq416; citation_id=CR2;'/>
    
    <meta name="citation_reference" content='International Cancer Genome Consortium Data Portal&#8211;a one-stop shop for cancer genomics data. Database : the journal of biological databases and curation. 2011, 2011 (0): bar026-[
                    http://dx.doi.org/10.1093/database/bar026
                    
                  ].'/>
    
    <meta name="citation_reference" content='citation_journal_title=Science; citation_title=On the future of genomic data; citation_author=SD Kahn; citation_volume=331; citation_issue=6018; citation_publication_date=2011; citation_pages=728-729; citation_doi=10.1126/science.1197891; citation_id=CR4;'/>
    
    <meta name="citation_reference" content='citation_journal_title=PLoS Comput Biol; citation_title=Biomedical cloud computing with amazon web services; citation_author=VA Fusaro; citation_author=P Patil; citation_author=E Gafni; citation_author=DP Wall; citation_author=PJ Tonellato; citation_volume=7; citation_issue=8; citation_publication_date=2011; citation_pages=1-6; citation_doi=10.1371/journal.pcbi.1002147; citation_id=CR5;'/>
    
    <meta name="citation_reference" content='citation_journal_title=Nat Biotechnol; citation_title=Cloud computing and the DNA data race; citation_author=VA Fusaro; citation_author=P Patil; citation_author=E Gafni; citation_author=DP Wall; citation_author=PJ Tonellato; citation_volume=28; citation_issue=7; citation_publication_date=2010; citation_pages=691-693; citation_doi=10.1038/nbt0710-691; citation_id=CR6;'/>
    
    <meta name="citation_reference" content='citation_journal_title=Genome Biol; citation_title=The case for cloud computing in genome informatics; citation_author=VA Fusaro; citation_author=P Patil; citation_author=E Gafni; citation_author=DP Wall; citation_author=PJ Tonellato; citation_volume=11; citation_issue=5; citation_publication_date=2010; citation_pages=207; citation_doi=10.1186/gb-2010-11-5-207; citation_id=CR7;'/>
    
    <meta name="citation_reference" content='citation_journal_title=Nat Rev Genet; citation_title=Big data, but are we ready?; citation_author=O Trelles; citation_author=P Prins; citation_author=M Snir; citation_author=RC Jansen; citation_volume=12; citation_issue=3; citation_publication_date=2011; citation_pages=224; citation_doi=10.1038/nrg2857-c1; citation_id=CR8;'/>
    
    <meta name="citation_reference" content='citation_journal_title=Science; citation_title=Will computers crash genomics?; citation_author=E Pennisim; citation_volume=331; citation_issue=6018; citation_publication_date=2011; citation_pages=666-668; citation_doi=10.1126/science.331.6018.666; citation_id=CR9;'/>
    
    <meta name="citation_reference" content='citation_journal_title=Comput Sci Rev; citation_title=Textual data compression in computational biology: algorithmic techniques; citation_author=R Giancarlo; citation_author=D Scaturro; citation_author=F Utro; citation_volume=6; citation_issue=1; citation_publication_date=2012; citation_pages=1-25; citation_doi=10.1016/j.cosrev.2011.11.001; citation_id=CR10;'/>
    
    <meta name="citation_reference" content='citation_journal_title=Entropy; citation_title=Data compression concepts and algorithms and their applications to bioinformatics; citation_author=&#214;U Nalbantoglu; citation_author=DJ Russell; citation_author=K Sayood; citation_volume=12; citation_publication_date=2010; citation_pages=34-52; citation_doi=10.3390/e12010034; citation_id=CR11;'/>
    
    <meta name="citation_reference" content='citation_title=Compressing biological sequences using self adjusting data structures; citation_inbook_title=10th IEEE International Conference on Information Technology and Applications in Biomedicine; citation_author=D Antoniou; citation_author=E Theodoridis; citation_author=A Tsakalidis; citation_publication_date=2010; citation_pages=1-5; citation_id=CR12;'/>
    
    <meta name="citation_reference" content='citation_title=Compressing the Human Genome Using Exclusively Markov Models; citation_inbook_title=PACBB, Volume 93 of Advances in Intelligent and Soft Computing; citation_author=D Pratas; citation_author=AJ Pinho; citation_publication_date=2011; citation_pages=213-220; citation_id=CR13;'/>
    
    <meta name="citation_reference" content='citation_journal_title=Bioinformatics; citation_title=Robust relative compression of genomes with random access; citation_author=S Deorowicz; citation_author=S Grabowski; citation_volume=27; citation_issue=21; citation_publication_date=2011; citation_pages=2979-2986; citation_doi=10.1093/bioinformatics/btr505; citation_id=CR14;'/>
    
    <meta name="citation_reference" content='citation_title=Relative Lempel-Ziv compression of genomes for large-scale storage and retrieval; citation_inbook_title=String Processing and Information Retrieval, Volume 6393 of Lecture Notes in Computer Science; citation_author=S Kuruppu; citation_author=S Puglisi; citation_author=J Zobel; citation_publisher=Springer; citation_publication_date=2010; citation_pages=201-206; citation_id=CR15;'/>
    
    <meta name="citation_reference" content='citation_title=Relative Lempel-Ziv compression of genomes for large-scale storage and retrieval; citation_inbook_title=Proceedings of the 17th international conference on String processing and information retrieval, SPIRE&#8217;10; citation_author=S Kuruppu; citation_author=SJ Puglisi; citation_author=J Zobel; citation_publisher=Springer-Verlag; citation_publication_date=2010; citation_pages=201-206; citation_id=CR16;'/>
    
    <meta name="citation_reference" content='citation_journal_title=Algorithmica; citation_title=On-line construction of suffix trees; citation_author=E Ukkonen; citation_volume=14; citation_publication_date=1995; citation_pages=249-260; citation_doi=10.1007/BF01206331; citation_id=CR17;'/>
    
    <meta name="citation_reference" content='citation_title=CST++; citation_inbook_title=String Processing and Information Retrieval -17th International Symposium, SPIRE 2010; citation_author=E Ohlebusch; citation_author=J Fischer; citation_author=S Gog; citation_publication_date=2010; citation_pages=322-333; citation_id=CR18;'/>
    
    <meta name="citation_reference" content='citation_journal_title=IEEE/ACM Trans Comput Biol Bioinformatics; citation_title=Iterative dictionary construction for compression of large DNA data sets; citation_author=S Kuruppu; citation_author=B Beresford-Smith; citation_author=T Conway; citation_author=J Zobel; citation_volume=9; citation_publication_date=2012; citation_pages=137-149; citation_doi=10.1109/TCBB.2011.82; citation_id=CR19;'/>
    
    <meta name="citation_reference" content='citation_title=A simple statistical algorithm for biological sequence compression; citation_inbook_title=Proceedings of the 2007 Data Compression Conference; citation_author=M Duc Cao; citation_author=TI Dix; citation_author=L Allison; citation_author=C Mears; citation_publisher=IEEE Computer Society,; citation_publication_date=2007; citation_pages=43-52; citation_id=CR20;'/>
    
    <meta name="citation_reference" content='citation_journal_title=Bioinformatics; citation_title=Human genomes as email attachments; citation_author=S Christley; citation_author=Y Lu; citation_author=C Li; citation_author=X Xie; citation_volume=25; citation_issue=2; citation_publication_date=2009; citation_pages=274-275; citation_doi=10.1093/bioinformatics/btn582; citation_id=CR21;'/>
    
    <meta name="citation_reference" content='citation_title=Engineering relative compression of genomes; citation_inbook_title=ArXiv; citation_author=S Grabowski; citation_author=S Deorowicz; citation_publication_date=2011,; citation_id=CR22;'/>
    
    <meta name="citation_reference" content='citation_title=LZ77-like compression with fast random access; citation_inbook_title=Proceedings of the 2010 Data Compression Conference, DCC &#8217;10; citation_author=S Kreft; citation_author=G Navarro; citation_publisher=IEEE Computer Society,; citation_publication_date=2010; citation_pages=239-248; citation_id=CR23;'/>
    
    <meta name="citation_reference" content='citation_title=Compression of DNA sequences; citation_inbook_title=Data Compression Conference; citation_author=S Grumbach; citation_author=F Tahi; citation_publication_date=1993; citation_pages=340-350; citation_id=CR24;'/>
    
    <meta name="citation_reference" content='citation_title=A compression algorithm for DNA sequences and its applications in genome comparison; citation_inbook_title=Proceedings of the fourth annual international conference on Computational molecular biology, RECOMB &#8217;00; citation_author=X Chen; citation_author=S Kwong; citation_author=M Li; citation_publisher=ACM; citation_publication_date=2000; citation_pages=107-107; citation_id=CR25;'/>
    
    <meta name="citation_reference" content='citation_journal_title=Software - Practice and Experience; citation_title=A simple and fast DNA compressor; citation_author=G Manzini; citation_author=M Rastero; citation_volume=34; citation_publication_date=2004; citation_pages=1397-1411; citation_doi=10.1002/spe.619; citation_id=CR26;'/>
    
    <meta name="citation_reference" content='citation_title=DNA compression challenge revisited: a dynamic programming approach; citation_inbook_title=Combinatorial Pattern Matching, Volume 3537 of Lecture Notes in Computer Science; citation_author=B Behzadi; citation_author=F Le Fessant; citation_publisher=Springer; citation_publication_date=2005; citation_pages=85-96; citation_id=CR27;'/>
    
    <meta name="citation_reference" content='citation_inbook_title=Grammar-based compression of DNA sequences; citation_author=N Cherniavsky; citation_author=R Ladner; citation_publication_date=2004; citation_id=CR28;'/>
    
    <meta name="citation_reference" content='citation_journal_title=Data Compression Conference; citation_title=A simple statistical algorithm for biological sequence compression; citation_author=MD Cao; citation_author=TI Dix; citation_author=L Allison; citation_author=C Mears; citation_volume=0; citation_publication_date=2007; citation_pages=43-52; citation_id=CR29;'/>
    
    <meta name="citation_reference" content='citation_journal_title=Genome Informatics; citation_title=Biological sequence compression algorithms; citation_author=T Matsumoto; citation_author=K Sadakane; citation_author=H Imai; citation_volume=11; citation_publication_date=2000; citation_pages=43-52; citation_id=CR30;'/>
    
    <meta name="citation_reference" content='citation_journal_title=PLoS ONE; citation_title=Improving transmission efficiency of large sequence alignment/map (SAM) files; citation_author=MN Sakib; citation_author=J Tang; citation_author=WJ Zheng; citation_author=CT Huang; citation_volume=6; citation_issue=12; citation_publication_date=2011; citation_pages=e28251; citation_doi=10.1371/journal.pone.0028251; citation_id=CR31;'/>
    
    <meta name="citation_reference" content='citation_journal_title=Genome Res; citation_title=Efficient storage of high throughput sequencing data using reference-based compression; citation_author=M Hsi-Yang Fritz; citation_author=R Leinonen; citation_author=G Cochrane; citation_author=E Birney; citation_volume=21; citation_issue=5; citation_publication_date=2011; citation_pages=734-740; citation_doi=10.1101/gr.114819.110; citation_id=CR32;'/>
    
    <meta name="citation_reference" content='BOOST C++ Libraries. [
                    http://www.boost.org
                    
                  ].]'/>
    
    <meta name="citation_reference" content='citation_title=CST++; citation_inbook_title=SPIRE&#8217;10; citation_author=E Ohlebusch; citation_author=J Fischer; citation_author=S Gog; citation_publication_date=2010; citation_pages=322-333; citation_id=CR34;'/>
    
    <meta name="citation_reference" content='citation_journal_title=Nature; citation_title=A map of human genome variation from population-scale sequencing; citation_volume=467; citation_issue=7319; citation_publication_date=2010; citation_pages=1061-1073; citation_doi=10.1038/nature09534; citation_id=CR35;'/>
    
    <meta name="citation_reference" content='citation_journal_title=Bioinformatics (Oxford, England); citation_title=The variant call format and VCFtools; citation_author=P Danecek; citation_author=A Auton; citation_author=G Abecasis; citation_author=CA Albers; citation_author=E Banks; citation_author=MA DePristo; citation_author=RE Handsaker; citation_author=G Lunter; citation_author=GT Marth; citation_author=ST Sherry; citation_author=G McVean; citation_author=R Durbin; citation_volume=27; citation_issue=15; citation_publication_date=2011; citation_pages=2156-2158; citation_doi=10.1093/bioinformatics/btr330; citation_id=CR36;'/>
    
    <meta name="citation_reference" content='citation_journal_title=Genome Res; citation_title=The human genome browser at UCSC; citation_author=WJ Kent; citation_author=CW Sugnet; citation_author=TS Furey; citation_author=KM Roskin; citation_author=TH Pringle; citation_author=AM Zahler; citation_author=D Haussler; citation_volume=12; citation_issue=6; citation_publication_date=2002; citation_pages=996-1006; citation_doi=10.1101/gr.229102. Article published online before print in May 2002; citation_id=CR37;'/>
    

    <meta charset="UTF-8"/>
<meta http-equiv="X-UA-Compatible" content="IE=edge"/>
<meta name="viewport" content="width=device-width, initial-scale=1"/>

    <link rel="apple-touch-icon" sizes="57x57"
          href="/static/images/apple-touch-icon-57x57.e249e00d.png"/>
    <link rel="apple-touch-icon" sizes="60x60"
          href="/static/images/apple-touch-icon-60x60.fb82a36f.png"/>
    <link rel="apple-touch-icon" sizes="72x72"
          href="/static/images/apple-touch-icon-72x72.e911dece.png"/>
    <link rel="apple-touch-icon" sizes="76x76"
          href="/static/images/apple-touch-icon-76x76.63bd2b8b.png"/>
    <link rel="apple-touch-icon" sizes="114x114"
          href="/static/images/apple-touch-icon-114x114.00d3da59.png"/>
    <link rel="apple-touch-icon" sizes="120x120"
          href="/static/images/apple-touch-icon-120x120.189ccdbf.png"/>
    <link rel="apple-touch-icon" sizes="144x144"
          href="/static/images/apple-touch-icon-144x144.4b231fd9.png"/>
    <link rel="apple-touch-icon" sizes="152x152"
          href="/static/images/apple-touch-icon-152x152.852d1454.png"/>
    <link rel="apple-touch-icon" sizes="180x180"
          href="/static/images/apple-touch-icon-180x180.2cfd7af9.png"/>
    <link rel="icon" type="image/png"
          href="/static/images/android-chrome-192x192.408ad37d.png" sizes="192x192"/>
    <link rel="icon" type="image/png" href="/static/images/favicon-16x16.51659b44.png"
          sizes="16x16"/>
    <link rel="icon" type="image/png" href="/static/images/favicon-32x32.44498c19.png"
          sizes="32x32"/>
    <link rel="icon" type="image/png" href="/static/images/favicon-96x96.08ca36c9.png"
          sizes="96x96"/>
    <link rel="icon" type="image/png" href="/static/images/favicon-194x194.0969f3a3.png"
          sizes="194x194"/>
    <link rel="manifest" href="/static/images/favicons/biomedcentral/manifest.json"/>
    <link rel="shortcut icon" href="/static/images/favicons/biomedcentral/favicon.ico"/>
    <meta name="msapplication-TileColor" content="#da532c"/>
    <meta name="msapplication-config"
          content="/static/images/favicons/biomedcentral/browserconfig.xml"/>
    <meta name="msapplication-TileImage"
          content="/static/images/mstile-144x144.8b117caf.png"/>
    <meta name="theme-color" content="#ffffff"/>


<script>(function(H){H.className=H.className.replace(/\bno-js\b/,'js')})(document.documentElement)</script>
<script>(function(d) {if (sessionStorage.getItem('webfonts') == 'true'){d.className += ' webfonts-loaded';}})(document.documentElement);</script>

    <link rel="stylesheet" media="screen" href="/static/css/basic.06b5e140.css"/>
    <link rel="stylesheet" media="print" href="/static/css/biomedcentral-print.12735a02.css"/>
    
    <link rel="stylesheet" class="js-ctm" href="/static/css/biomedcentral.299e619f.css"
          media="only screen and (-webkit-min-device-pixel-ratio:0) and (min-color-index:0), (-ms-high-contrast: none), only all and (min--moz-device-pixel-ratio:0) and (min-resolution: 3e1dpcm)">


<script>var _gaq = [];</script>

    <script type="text/javascript">
        config = {
            env: 'live',
            site: 'almob.biomedcentral.com',
            siteWithPath: 'almob.biomedcentral.com' + window.location.pathname,
            twitterHashtag: 'almob',
            cmsPrefix: 'http://studio-cms.springer.com/studio/',
            
            doi: '10.1186/1748-7188-7-30',
            
            publisherBrand: 'BioMed Central'
        };
    </script>


    <script>
        (function (i, s, o, g, r, a, m) {
            i['GoogleAnalyticsObject'] = r;
            i[r] = i[r] || function () {
                        (i[r].q = i[r].q || []).push(arguments)
                    }, i[r].l = 1 * new Date();
            a = s.createElement(o),
                    m = s.getElementsByTagName(o)[0];
            a.async = 1;
            a.src = g;
            m.parentNode.insertBefore(a, m)
        })(window, document, 'script', '//www.google-analytics.com/analytics.js', 'ga');

        var sampleRate = 20;

            
            ga('create', "UA-62059650-7",
                    {
                        cookieDomain: 'none',
                        allowLinker: true,
                        siteSpeedSampleRate: sampleRate
                    });
            ga("set", "anonymizeIp", true);
            ga('send', 'pageview');
            

        ga('create', "UA-54492316-9",
                {
                    cookieDomain: 'none',
                    allowLinker: true,
                    siteSpeedSampleRate: sampleRate,
                    name: 'global'
                });

        ga("global.set", "anonymizeIp", true);
        ga('global.send', 'pageview');
    </script>



    
    <link rel="canonical" href="//almob.biomedcentral.com/articles/10.1186/1748-7188-7-30"/>
    


    
    <meta property="og:url" content="http://almob.biomedcentral.com/articles/10.1186/1748-7188-7-30"/>
    <meta property="og:type" content="article"/>
    <meta property="og:site_name" content="Algorithms for Molecular Biology"/>
    <meta property="og:title" content="Adaptive efficient compression of genomes"/>
    <meta property="og:description" content="Modern high-throughput sequencing technologies are able to generate DNA sequences at an ever increasing rate. In parallel to the decreasing experimental time and cost necessary to produce DNA sequences, computational requirements for analysis and storage of the sequences are steeply increasing. Compression is a key technology to deal with this challenge. Recently, referential compression schemes, storing only the differences between a to-be-compressed input and a known reference sequence, gained a lot of interest in this field. However, memory requirements of the current algorithms are high and run times often are slow. In this paper, we propose an adaptive, parallel and highly efficient referential sequence compression method which allows fine-tuning of the trade-off between required memory and compression speed. When using 12 MB of memory, our method is for human genomes on-par with the best previous algorithms in terms of compression ratio (400:1) and compression speed. In contrast, it compresses a complete human genome in just 11 seconds when provided with 9 GB of main memory, which is almost three times faster than the best competitor while using less main memory."/>
    <meta property="og:image" content="https://images.springer.com/sgw/journals/medium/13015.jpg"/>

</head>
<body class="fulltext">
    <div class="container" style="height: 0"><div id="wt_refpoint"></div></div>
    <p class="skip-to-main">
    <a class="skip-to-main__link" href="#main">Skip to main content</a>
</p>

    <div id="top">
        
<div id="RealMediaLeaderboard" class="adsbox RealMediaAd Ad Ad--leaderboard" data-position="Top">
    <div class="Ad_content">
        <p class="Ad_label">Advertisement</p>
        
    </div>
</div>


         
            <header class="header" role="banner">
                <div class="header__inner">
                    <a id="header-logo" href="//www.biomedcentral.com/" class="logo header__logo" title="BioMed Central">
                        <img srcset="/static/images/biomed-central-logo.23470c06.svg" src="/static/images/biomed-central-logo.5300b479.png" class="logo__img" alt="Biomed Central" />
                    </a>
                    <div class="header__right">
                        <div class="toggles js-fade-toggle">
                            <p id="loginLink" class="sign-in toggles__item" data-replace="true" data-placeholder="/placeholder/v1/username">
    <a class="sign-in__link" href="//www.biomedcentral.com/my-account" data-track-event="click" data-event-category="Account" data-event-action="Click My Account">
        <span class="sign-in__no-js-account-hide">Login<span class="sign-in__js-account-hide"> to </span></span><span class="sign-in__js-account-hide sign-in__no-js-capitalise">your account</span>
    </a>
</p>
                            
                            <a href="#globalSearch" class="toggles__item toggles__item--search" data-toggle="search"><span class="sr-only">Search</span></a>
                        </div>
                        <form role="search" id="globalSearch" class="search search--expandable header__search" method="GET" action="//www.biomedcentral.com/search" data-track-event="submit"
      data-event-category="Search and Results" data-event-action="Submit search" data-event-label="">
    <div class="search__content">
        <label for="publisherSearch" class="sr-only">Search BioMed Central articles</label>
        <input id="publisherSearch" class="search__input" autocomplete="off" role="textbox" name="query" type="text" value="" title="Search BioMed Central" placeholder="Search BioMed Central articles"/>
        <button id="search" class="search__button" type="submit">
            <span class="sr-only">Search</span>
        </button>
        <input type="hidden" name="searchType" value="publisherSearch"/>
    </div>
</form>

                    </div>
                    
                </div>
            </header>
            <div id="membership-message-loader-desktop" class="placeholder" data-placeholder="/placeholder/v1/membership/message"></div>
        

        <main role="main">
            
            <header class="journal-header" role="banner">
                <div class="journal-header__inner">
                    <h2 class="journal-header__title journal-header__title--medium" id="journalTitle">
                        <a class="journal-header__link" href="/">Algorithms for Molecular Biology</a>
                    </h2>
                </div>
                <div class="navbar">
                    <div class="navbar__inner">
                        

                        
                        <nav class="navbar__nav">
                            <h3 class="navbar__menu-toggle" data-toggle="collapse" data-target="journal-menu">
                                <span class="sr-only">Main </span>menu
                            </h3>
                            <ul id="journal-menu" class="navbar__menu" id="siteNavigation" role="menu">
                                
                                    <li class="navbar__item" role="menuitem">
                                        <a class="navbar__link" href='/'>Home</a>
                                    </li>
                                
                                    <li class="navbar__item" role="menuitem">
                                        <a class="navbar__link" href='/about'>About</a>
                                    </li>
                                
                                    <li class="navbar__item" role="menuitem">
                                        <a class="navbar__link is-active" href='/articles'>Articles</a>
                                    </li>
                                
                                    <li class="navbar__item" role="menuitem">
                                        <a class="navbar__link" href='/submission-guidelines'>Submission Guidelines</a>
                                    </li>
                                
                            </ul>
                        </nav>
                        
                    </div>
                </div>
                <div class="journal-identity journal-identity--globules-red">
                    
                </div>
            </header>
        
            <div class="container">
                
                <div id="main" class="grid">
                
                    <div class="main__content grid__col grid__col--xs-12 grid__col--md-9" id="main-content">
                        <div class="block" id="Test-ImgSrc">
                            
                            
                            
    <div class="FulltextWrapper">
        <div class="Categories">
            <div class="ArticleCategory">Research</div>
            <div class="OpenAccessLabel">Open Access</div>
            
        </div>
        
    </div>

                            
                            <div class="FulltextWrapper"><div xmlns="" xmlns:fn="http://www.w3.org/2005/xpath-functions" xmlns:meta="http://www.springer.com/app/meta" class="MainTitleSection"><h1 xmlns="http://www.w3.org/1999/xhtml" class="ArticleTitle" lang="en">Adaptive efficient compression of genomes</h1></div><div xmlns="" xmlns:fn="http://www.w3.org/2005/xpath-functions" xmlns:meta="http://www.springer.com/app/meta" class="AuthorNames"><ul class="u-listReset"><li class="Author" data-jumpto="Aff"><span xmlns="http://www.w3.org/1999/xhtml" class="AuthorName">Sebastian&nbsp;Wandelt</span><sup><a href="#Aff1" class="AffiliationID">1</a></sup><a href="mailto:wandelt@informatik.hu-berlin.de" class="EmailAuthor" title="Email author"><span class="ContactIcon" aria-hidden="true"></span><span class="u-srOnly">Email author</span></a> and </li><li class="Author" data-jumpto="Aff"><span xmlns="http://www.w3.org/1999/xhtml" class="AuthorName">Ulf&nbsp;Leser</span><sup><a href="#Aff1" class="AffiliationID">1</a></sup></li></ul></div><div xmlns="" xmlns:fn="http://www.w3.org/2005/xpath-functions" xmlns:meta="http://www.springer.com/app/meta" class="ArticleContextInformation"><div class="ContextInformation"><div class="ContextInformationJournalTitles"><span class="ArticleCitation"><span xmlns="http://www.w3.org/1999/xhtml" class="JournalTitle">Algorithms for Molecular Biology</span><span class="ArticleCitation_Year">2012</span><span class="ArticleCitation_Volume"><strong>7</strong>:30</span></span></div><p class="ArticleDOI"><strong>DOI: </strong>10.1186/1748-7188-7-30</p><p class="Copyright">© &nbsp;Wandelt and Leser; licensee BioMed Central Ltd.&nbsp;2012</p></div></div><div class="ArticleHistory"><p class="History HistoryReceived"><strong>Received: </strong>25&nbsp;July&nbsp;2012</p><p class="History HistoryAccepted"><strong>Accepted: </strong>26&nbsp;October&nbsp;2012</p><p class="History HistoryOnlineDate"><strong>Published: </strong>12&nbsp;November&nbsp;2012</p></div></div>
                            
                            
                            
    

                            <div class="FulltextWrapper"><section xmlns="" xmlns:fn="http://www.w3.org/2005/xpath-functions" xmlns:meta="http://www.springer.com/app/meta" class="Abstract" id="Abs1" lang="en"><h2 class="Heading js-ToggleCollapseSection">Abstract</h2><div class="js-CollapseSection"><div xmlns="http://www.w3.org/1999/xhtml" class="AbstractSection" id="ASec1">
                <h3 xmlns="" class="Heading">Abstract</h3>
                <p class="Para">Modern high-throughput sequencing technologies are able to generate DNA sequences at an ever increasing rate. In parallel to the decreasing experimental time and cost necessary to produce DNA sequences, computational requirements for analysis and storage of the sequences are steeply increasing. Compression is a key technology to deal with this challenge. Recently, referential compression schemes, storing only the differences between a to-be-compressed input and a known reference sequence, gained a lot of interest in this field. However, memory requirements of the current algorithms are high and run times often are slow. In this paper, we propose an adaptive, parallel and highly efficient referential sequence compression method which allows fine-tuning of the trade-off between required memory and compression speed. When using 12 MB of memory, our method is for human genomes on-par with the best previous algorithms in terms of compression ratio (400:1) and compression speed. In contrast, it compresses a complete human genome in just 11 seconds when provided with 9 GB of main memory, which is almost three times faster than the best competitor while using less main memory.</p>
              </div><div xmlns="http://www.w3.org/1999/xhtml" class="KeywordGroup" lang="en">
              <h3 xmlns="" class="Heading">Keywords</h3>
              <span class="Keyword">Sequence compression</span>
              <span class="Keyword">Referential compression</span>
              <span class="Keyword">Heuristics</span>
              <span class="Keyword">Scalability</span>
            </div></div></section>
            <section xmlns="" xmlns:fn="http://www.w3.org/2005/xpath-functions" xmlns:meta="http://www.springer.com/app/meta" id="Sec1" class="Section1 RenderAsSection1"><h2 class="Heading js-ToggleCollapseSection">Background</h2><div class="js-CollapseSection"><p xmlns="http://www.w3.org/1999/xhtml" class="Para">The development of novel high-throughput DNA sequencing techniques has led to an ever increasing flood of data. While it took roughly 12 years and an estimated amount of 3 billion USD to decipher the first human genome[<span class="CitationRef"><a href="#CR1">1</a></span>], current techniques, usually summarized under the term second generation sequencing (SGS), are able to produce roughly the same amount of data in about a week at a current cost of roughly 2000 USD. On top, third generation sequencing promises to deliver a further speed-up, reducing the time and price for sequencing a human genome from weeks to days and from thousands to under a hundred USD, respectively[<span class="CitationRef"><a href="#CR2">2</a></span>].</p><p xmlns="http://www.w3.org/1999/xhtml" class="Para">Large-scale projects are generating comprehensive surveys of the genomic landscape of various diseases by sequencing thousands of genomes[<span class="CitationRef"><a href="#CR3">3</a></span>]. Managing, storing and analyzing this quickly growing amount of data is challenging[<span class="CitationRef"><a href="#CR4">4</a></span>]. It requires large disk arrays for storage, and large compute clusters for analysis. A recent suggestion is to use cloud infrastructures for this purpose[<span class="CitationRef"><a href="#CR5">5</a></span>–<span class="CitationRef"><a href="#CR7">7</a></span>]. However, before being analyzed in a cloud, data first has to be shipped to the cloud, making bandwidth in file transfer one of the major bottlenecks in cloud-based DNA analysis[<span class="CitationRef"><a href="#CR8">8</a></span>]. Accordingly, sequence compression is a key technology to cope with the increasing flood of DNA sequences[<span class="CitationRef"><a href="#CR9">9</a></span>–<span class="CitationRef"><a href="#CR11">11</a></span>].</p><p xmlns="http://www.w3.org/1999/xhtml" class="Para">To store a complete genome of a human being, one needs roughly 3GB (uncompressed). Substitutional or statistic compression schemes can reduce the space requirements by up to 6:1 (one base is encoded with up to 1.3 Bit)[<span class="CitationRef"><a href="#CR12">12</a></span>, <span class="CitationRef"><a href="#CR13">13</a></span>]. However, in many projects only genomes from one species are considered. This means that projects often deal with hundreds of highly similar genomes; for instance, two randomly selected human genomes are identical to an estimated 99.9%. This observation is exploited by so-called referential compression schemes, which only encode the differences of an input sequence with respect to a pre-selected reference sequence. Using space-efficient encoding of differences and clever algorithms for finding long stretches of DNA without differences, the best current referential compression algorithm we are aware of reports a compression rates of up to 500:1 for human genomes[<span class="CitationRef"><a href="#CR14">14</a></span>].</p><p xmlns="http://www.w3.org/1999/xhtml" class="Para">However, all existing compression schemes have in common that they have very high demands on the underlying hardware (up to 25 GB, for instance[<span class="CitationRef"><a href="#CR15">15</a></span>, <span class="CitationRef"><a href="#CR16">16</a></span>]). Furthermore, the time needed for compressing the amount of sequences corresponding to a human genome may be up to several hours, which easily out-weights the time savings during file transfer. Furthermore, existing approaches cannot easily be parallelized to reduce compression time, because such a parallelization requires to share information about the generated compression model. This communication/synchronization overhead may easily mitigate the positive effects gained by parallelization.</p><p xmlns="http://www.w3.org/1999/xhtml" class="Para">In this paper, we present an adaptive, scalable and highly efficient algorithm for referential compression of DNA sequences genomes. It is able to gracefully trade-off compression time and space requirements while consistently achieving very good compression rates. For instance, our algorithms requires only 12 MB memory to compress/decompress a human genome at the same speed as the fastest existing approach. Using 9 GB of memory usage, our method performs up to three times faster than the best competitor while still needing less main memory. Both variants achieve similar compression rates of approximately 400:1 for human DNA. For yeast sequences, the compression ratios are lower, since two yeast genomes can have substantial differences.</p><p xmlns="http://www.w3.org/1999/xhtml" class="Para">The remaining part of this paper is structured as follows. We first explain the main ideas behind our algorithm in Section “General Idea”. The concrete design choices and algorithms are described in detail in Section “Genome Compression”. We discuss related work in Section “Related Work”, before we provide an evaluation of our method in Section “Evaluation”. The paper is concluded with Section “Conclusions”.</p></div></section>
            <section xmlns="" xmlns:fn="http://www.w3.org/2005/xpath-functions" xmlns:meta="http://www.springer.com/app/meta" id="Sec2" class="Section1 RenderAsSection1"><h2 class="Heading js-ToggleCollapseSection">General Idea</h2><div class="js-CollapseSection"><p xmlns="http://www.w3.org/1999/xhtml" class="Para">We denote strings with <em xmlns="" class="EmphasisTypeItalic">s,t</em>. The length of a string <em xmlns="" class="EmphasisTypeItalic">s</em> is denoted with |<em xmlns="" class="EmphasisTypeItalic">s</em>| and the substring starting at position <em xmlns="" class="EmphasisTypeItalic">i</em> with length <em xmlns="" class="EmphasisTypeItalic">n</em> is denoted <em xmlns="" class="EmphasisTypeItalic">s</em>(<em xmlns="" class="EmphasisTypeItalic">i,n</em>). <em xmlns="" class="EmphasisTypeItalic">s</em>(<em xmlns="" class="EmphasisTypeItalic">i</em>) is an abbreviation for <em xmlns="" class="EmphasisTypeItalic">s</em>(<em xmlns="" class="EmphasisTypeItalic">i</em>,1). All positions in a string are zero-based, i.e. the first character is accessed by <em xmlns="" class="EmphasisTypeItalic">s</em>(0). The concatenation of two strings <em xmlns="" class="EmphasisTypeItalic">s</em> and <em xmlns="" class="EmphasisTypeItalic">t</em> is denoted with <em xmlns="" class="EmphasisTypeItalic">s</em><span class="stix">∘</span><em xmlns="" class="EmphasisTypeItalic">t</em>. Although a genome can be encoded with four characters, i.e. A,C,G, and T, we allow arbitrary symbols. For instance, symbol <em xmlns="" class="EmphasisTypeItalic">N</em> is often used to indicate an unknown base. Given two strings <em xmlns="" class="EmphasisTypeItalic">s</em> and <em xmlns="" class="EmphasisTypeItalic">t</em>, the <em xmlns="" class="EmphasisTypeItalic">longest prefix-suffix match of</em> <em xmlns="" class="EmphasisTypeItalic">s</em> in <em xmlns="" class="EmphasisTypeItalic">t</em>, is the longest string <em xmlns="" class="EmphasisTypeItalic">t</em><sub>
                  <em xmlns="" class="EmphasisTypeItalic">m</em>
                </sub>, such that <em xmlns="" class="EmphasisTypeItalic">t</em> = <em xmlns="" class="EmphasisTypeItalic">t</em><sub>1</sub><span class="stix">∘</span><em xmlns="" class="EmphasisTypeItalic">t</em><sub>
                  <em xmlns="" class="EmphasisTypeItalic">m</em>
                </sub><span class="stix">∘</span><em xmlns="" class="EmphasisTypeItalic">t</em><sub>2</sub> and <em xmlns="" class="EmphasisTypeItalic">s</em>(0,|<em xmlns="" class="EmphasisTypeItalic">t</em><sub>
                  <em xmlns="" class="EmphasisTypeItalic">m</em>
                </sub>|) = <em xmlns="" class="EmphasisTypeItalic">t</em><sub>
                  <em xmlns="" class="EmphasisTypeItalic">m</em>
                </sub>.</p><p xmlns="http://www.w3.org/1999/xhtml" class="Para">We want to compress a given input genome with respect to a reference genome, by only encoding differences between the input and the reference. This yields loss-less compression, i.e. based on the reference genome and the difference description it is possible to recover the input genome. The idea is to split up a given reference genome into blocks. In general, two genomes of the same species are very similar to each other, and blocks are chosen in a way that long matches of to-be-compressed blocks can often be found in reference blocks by local search. Please note that splitting the reference into blocks of fixed length is not an optimal way, since different sizes of the blocks can provide better performance in terms of compression ratio. However, for the sake of compression speed and easier data handling, we only consider a fixed block length.</p><p xmlns="http://www.w3.org/1999/xhtml" class="Para">For each reference genome block, a suffix tree is computed. The suffix trees allow to find longest prefix-suffix matches of parts of the input genome and the reference genome efficiently. The compression process is informally summarized in Algorithm 1.</p><section id="Sec3" class="Section2 RenderAsSection2">
                <h3 class="Heading">Algorithm 1: Sketch of our Compression Algorithm</h3>
                <p xmlns="http://www.w3.org/1999/xhtml" class="Para">1: <strong xmlns="" class="EmphasisTypeBold">while</strong> Input contains characters</p>
                <p xmlns="http://www.w3.org/1999/xhtml" class="Para">2: Find matching reference block <em xmlns="" class="EmphasisTypeItalic">B</em> for current input position</p>
                <p xmlns="http://www.w3.org/1999/xhtml" class="Para">3: Perform referential compression with respect to <em xmlns="" class="EmphasisTypeItalic">B</em> until we cannot find “long” matches anymore</p>
                <p xmlns="http://www.w3.org/1999/xhtml" class="Para">4: <strong xmlns="" class="EmphasisTypeBold">end while</strong></p>
                <p xmlns="http://www.w3.org/1999/xhtml" class="Para">The compression algorithm generates a set of referential matches with respect to reference blocks. The output of our compression algorithm is a compressed file of entries, such that each entry is one of the following:</p>
                <div xmlns="http://www.w3.org/1999/xhtml" class="Para">
                  <div class="UnorderedList"><ul class="UnorderedListMarkBullet"><li>
                      <p class="Para"><strong xmlns="" class="EmphasisTypeBold">Block-change entry</strong> <em xmlns="" class="EmphasisTypeItalic">BC</em>(<em xmlns="" class="EmphasisTypeItalic">i</em>): next entries are encoded with respect to reference block <em xmlns="" class="EmphasisTypeItalic">i</em>.</p>
                    </li><li>
                      <p class="Para"><strong xmlns="" class="EmphasisTypeBold">Relative match entry</strong> <em xmlns="" class="EmphasisTypeItalic">RM</em>(<em xmlns="" class="EmphasisTypeItalic">i,j</em>): The input matches the reference block at position <em xmlns="" class="EmphasisTypeItalic">i</em> for <em xmlns="" class="EmphasisTypeItalic">j</em> characters.</p>
                    </li><li>
                      <p class="Para"><strong xmlns="" class="EmphasisTypeBold">Raw entry</strong> <em xmlns="" class="EmphasisTypeItalic">R</em>(<em xmlns="" class="EmphasisTypeItalic">s</em>): A string <em xmlns="" class="EmphasisTypeItalic">s</em> is encoded raw (for instance if there is no good matching block).</p>
                    </li></ul></div>
                </div>
                <div xmlns="http://www.w3.org/1999/xhtml" class="Para">An example compression is given in Figure<span class="InternalRef"><a href="#Fig1">1</a></span>. An input sequence “TACGTAAT..” is compressed with respect to a reference sequence “ACGACGTA..”. The first character of the input is encoded raw, with entry <em xmlns="" class="EmphasisTypeItalic">R</em>(<em xmlns="" class="EmphasisTypeItalic">T</em>), then the first reference block is chosen, with entry <em xmlns="" class="EmphasisTypeItalic">BC</em>(0) , in order to referentially encode the subsequence “ACGTA”, with entry <em xmlns="" class="EmphasisTypeItalic">RM</em>(3,5).<figure xmlns="" class="Figure" id="Fig1"><div xmlns="http://www.w3.org/1999/xhtml" class="MediaObject">
                      <img src="https://static-content.springer.com/image/art%3A10.1186%2F1748-7188-7-30/MediaObjects/13015_2012_Article_165_Fig1_HTML.jpg" alt="https://static-content.springer.com/image/art%3A10.1186%2F1748-7188-7-30/MediaObjects/13015_2012_Article_165_Fig1_HTML.jpg">
                    </div><figcaption class="Caption" lang="en"><div xmlns="http://www.w3.org/1999/xhtml" class="CaptionContent"><span class="CaptionNumber">Figure 1</span>
                        <p xmlns="" class="SimplePara"><strong class="EmphasisTypeBold">Example for relative compression.</strong> An example compression is given in Figure<span xmlns="http://www.w3.org/1999/xhtml" class="InternalRef"><a href="#Fig1">1</a></span>. An input sequence “TACGTAAT..” is compressed with respect to a reference sequence “ACGACGTA..”. The first character of the input is encoded raw, with entry <em class="EmphasisTypeItalic">R</em>(<em class="EmphasisTypeItalic">T</em>), then the first reference block is chosen, with entry <em class="EmphasisTypeItalic">BC</em>(0) , in order to referentially encode the subsequence “ACGTA”, with entry <em class="EmphasisTypeItalic">RM</em>(3,5).</p>
                      </div></figcaption></figure></div>
                <p xmlns="http://www.w3.org/1999/xhtml" class="Para">Please note that block change entries can be actually avoided, since local positions in a block can be easily transformed into global positions in the reference. In the remaining part, we will still assume the usage of block change entries.</p>
              </section></div></section>
            <section xmlns="" xmlns:fn="http://www.w3.org/2005/xpath-functions" xmlns:meta="http://www.springer.com/app/meta" id="Sec4" class="Section1 RenderAsSection1"><h2 class="Heading js-ToggleCollapseSection">Genome Compression</h2><div class="js-CollapseSection"><section id="Sec5" class="Section2 RenderAsSection2">
                <h3 class="Heading">Index Construction</h3>
                <p xmlns="http://www.w3.org/1999/xhtml" class="Para">In order to find matches of the input in reference blocks, we need an index structure over the reference blocks. Suffix trees[<span class="CitationRef"><a href="#CR17">17</a></span>] are tree data structures which allow for fast access to all suffixes of a given string. Each suffix is usually represented by a path from the root of the tree to a leaf. Please note that the longest prefix-suffix match problem of <em xmlns="" class="EmphasisTypeItalic">s</em> in <em xmlns="" class="EmphasisTypeItalic">t</em> can be be easily solved by depth-first search given a suffix tree for the string <em xmlns="" class="EmphasisTypeItalic">t</em>. The focus recently shifted towards so-called <em xmlns="" class="EmphasisTypeItalic">compressed suffix trees</em>. One example of a compressed suffix tree implementation is CST++[<span class="CitationRef"><a href="#CR18">18</a></span>], which uses a new approach to compress the longest common prefix-array and achieves, depending on the sampling rates of the (inverse) suffix array, an usual space overhead of (4−6)<span class="stix">∗</span><em xmlns="" class="EmphasisTypeItalic">n</em> bits. In the following, the compressed suffix tree of a string <em xmlns="" class="EmphasisTypeItalic">s</em> is denoted with <em xmlns="" class="EmphasisTypeItalic">CST</em>(<em xmlns="" class="EmphasisTypeItalic">s</em>).</p>
                <p xmlns="http://www.w3.org/1999/xhtml" class="Para">In addition, we need a reference genome for our compression algorithm. The reference sequence is given as a set of compressed FASTA-files, one for each chromosome. The index generation algorithm iterates over all chromosomes of the reference genome. Each chromosome is split up into blocks of a maximum size <em xmlns="" class="EmphasisTypeItalic">BS</em>. For instance, given the textual representation <em xmlns="" class="EmphasisTypeItalic">s</em><sub>1</sub> of Chromosome 1, we compute <em xmlns="" class="EmphasisTypeItalic">m</em> substrings <em xmlns="" class="EmphasisTypeItalic">b</em><sub>1,1</sub>,…,<em xmlns="" class="EmphasisTypeItalic">b</em><sub>1,<em xmlns="" class="EmphasisTypeItalic">m</em></sub>, such that</p>
                <div xmlns="http://www.w3.org/1999/xhtml" class="Para">
                  <div class="UnorderedList"><ul class="UnorderedListMarkBullet"><li>
                      <p class="Para"><em xmlns="" class="EmphasisTypeItalic">s</em><sub>1</sub> = <em xmlns="" class="EmphasisTypeItalic">b</em><sub>1,1</sub>,…,<em xmlns="" class="EmphasisTypeItalic">b</em><sub>1,<em xmlns="" class="EmphasisTypeItalic">m</em></sub>,</p>
                    </li><li>
                      <p class="Para"><span class="stix">∀</span><em xmlns="" class="EmphasisTypeItalic">i</em> ≤ (<em xmlns="" class="EmphasisTypeItalic">m</em>−1):|<em xmlns="" class="EmphasisTypeItalic">b</em><sub>1,<em xmlns="" class="EmphasisTypeItalic">i</em></sub>| = <em xmlns="" class="EmphasisTypeItalic">BS</em>, and</p>
                    </li><li>
                      <p class="Para">[<em xmlns="" class="EmphasisTypeItalic">b</em><sub>1,<em xmlns="" class="EmphasisTypeItalic">m</em></sub>] ≤ <em xmlns="" class="EmphasisTypeItalic">BS</em>.</p>
                    </li></ul></div>
                </div>
                <p xmlns="http://www.w3.org/1999/xhtml" class="Para">We do the same for all other chromosomes of the reference genome. For each reference block a compressed suffix tree is computed and stored on hard disk together with the raw reference block. In addition, we keep track of the order of the blocks, which is induced by their position on the chromosomes. This meta information is used below for optimizing our compression algorithm.</p>
                <p xmlns="http://www.w3.org/1999/xhtml" class="Para">The memory consumption during index creation is limited as follows: at each step of the index generation we have one raw reference block of size at most BS bytes in main memory plus (roughly) 4<span class="stix">∗</span>BS bytes for its compressed suffix tree. For example, given a BS of 4 MB, the main memory usage can be restricted to approximately 20 MB. The maximum value for BS is 300 MB, since the largest human chromosome (Chromosome 1) has about 247 million nucleotide base pairs, and each base is encoded as one byte.</p>
                <p xmlns="http://www.w3.org/1999/xhtml" class="Para">Please note that we could have encoded bases of the reference genome with 3 bits (five entries: A,C,G,N,T), in order to further reduce memory usage. However, our tests indicate that the space saving yields a significant time overhead for accessing bases. Furthermore, the 3-Bit-encoding would not allow us to compress sequences against references with other symbols than these five.</p>
              </section><section id="Sec6" class="Section2 RenderAsSection2">
                <h3 class="Heading">Compression and Decompression</h3>
                <p xmlns="http://www.w3.org/1999/xhtml" class="Para">In the following, we present our compression algorithm for genome sequences in detail. Algorithms do not show range checks for the sake of readability. Algorithm 2 assumes the input genome in <em xmlns="" class="EmphasisTypeItalic">Input</em> (as a string of bases). The input string is traversed from left to right, and depending on the current characters in the input and in the reference block, different subroutines are executed.</p>
              </section><section id="Sec7" class="Section2 RenderAsSection2">
                <h3 class="Heading">Algorithm 2: Compression Algorithm</h3>
                <p xmlns="http://www.w3.org/1999/xhtml" class="Para">1: <em xmlns="" class="EmphasisTypeItalic">P</em><sub>
                    <em xmlns="" class="EmphasisTypeItalic">in</em>
                  </sub>←0</p>
                <p xmlns="http://www.w3.org/1999/xhtml" class="Para">2: <em xmlns="" class="EmphasisTypeItalic">P</em><sub>
                    <em xmlns="" class="EmphasisTypeItalic">raw</em>
                  </sub>←0</p>
                <p xmlns="http://www.w3.org/1999/xhtml" class="Para">3: FIND-MATCH</p>
                <p xmlns="http://www.w3.org/1999/xhtml" class="Para">4: <strong xmlns="" class="EmphasisTypeBold">while</strong> <em xmlns="" class="EmphasisTypeItalic">P</em><sub>
                    <em xmlns="" class="EmphasisTypeItalic">in</em>
                  </sub>≠ <span class="stix">∣</span><em xmlns="" class="EmphasisTypeItalic">Input</em><span class="stix">∣</span><strong xmlns="" class="EmphasisTypeBold">do</strong></p>
                <p xmlns="http://www.w3.org/1999/xhtml" class="Para">5: <strong xmlns="" class="EmphasisTypeBold">if</strong> <em xmlns="" class="EmphasisTypeItalic">Input</em>(<em xmlns="" class="EmphasisTypeItalic">P</em><sub>
                    <em xmlns="" class="EmphasisTypeItalic">in</em>
                  </sub>) = <em xmlns="" class="EmphasisTypeItalic">B</em>(<em xmlns="" class="EmphasisTypeItalic">P</em><sub>
                    <em xmlns="" class="EmphasisTypeItalic">raw</em>
                  </sub>) <strong xmlns="" class="EmphasisTypeBold">then</strong></p>
                <p xmlns="http://www.w3.org/1999/xhtml" class="Para">6: ENCODE-REF</p>
                <p xmlns="http://www.w3.org/1999/xhtml" class="Para">7: <strong xmlns="" class="EmphasisTypeBold">else if</strong> <em xmlns="" class="EmphasisTypeItalic">Input</em>(<em xmlns="" class="EmphasisTypeItalic">P</em><sub>
                    <em xmlns="" class="EmphasisTypeItalic">in</em>
                  </sub>)<span class="stix">∉</span>{<em xmlns="" class="EmphasisTypeItalic">A</em>,<em xmlns="" class="EmphasisTypeItalic">C</em>,<em xmlns="" class="EmphasisTypeItalic">G</em>,<em xmlns="" class="EmphasisTypeItalic">T</em>} <strong xmlns="" class="EmphasisTypeBold">then</strong></p>
                <p xmlns="http://www.w3.org/1999/xhtml" class="Para">8: ENCODE-RAW</p>
                <p xmlns="http://www.w3.org/1999/xhtml" class="Para">9: <strong xmlns="" class="EmphasisTypeBold">else</strong></p>
                <p xmlns="http://www.w3.org/1999/xhtml" class="Para">10: FIND-MATCH</p>
                <p xmlns="http://www.w3.org/1999/xhtml" class="Para">11: <strong xmlns="" class="EmphasisTypeBold">end if</strong></p>
                <p xmlns="http://www.w3.org/1999/xhtml" class="Para">12: <strong xmlns="" class="EmphasisTypeBold">end while</strong></p>
                <p xmlns="http://www.w3.org/1999/xhtml" class="Para">In the beginning of the compression algorithm, a match for the current input position in the reference is searched, using function FIND-MATCH, Algorithm 3, explained in detail below. If the current reference block <em xmlns="" class="EmphasisTypeItalic">B</em> matches the input at the current position, then we try to generate a (as long as possible) reference match in function ENCODE-REF, Algorithm 4. If the current input base does not equal the current reference block base, and the input is not a <em xmlns="" class="EmphasisTypeItalic">normal</em> base, i.e. neither <em xmlns="" class="EmphasisTypeItalic">A</em>, <em xmlns="" class="EmphasisTypeItalic">C</em>, <em xmlns="" class="EmphasisTypeItalic">G</em>, or <em xmlns="" class="EmphasisTypeItalic">T</em>, then the base and all the following non-normal bases are added as raw entries to the compressed genome file in function ENCODE-RAW, Algorithm 5. Finally, if neither of the conditions is satisfied, then the algorithm tries to find a new match (either in the current block or in another reference block) using function FIND-MATCH.</p>
              </section><section id="Sec8" class="Section2 RenderAsSection2">
                <h3 class="Heading">Algorithm 3: FIND-MATCH Function</h3>
                <p xmlns="http://www.w3.org/1999/xhtml" class="Para">1: <em xmlns="" class="EmphasisTypeItalic">Max</em>←25</p>
                <p xmlns="http://www.w3.org/1999/xhtml" class="Para">2: <strong xmlns="" class="EmphasisTypeBold">if</strong> not HAS-LOCAL-ALTERNATIVE() <strong xmlns="" class="EmphasisTypeBold">then</strong></p>
                <p xmlns="http://www.w3.org/1999/xhtml" class="Para">3: <strong xmlns="" class="EmphasisTypeBold">for all</strong> <em xmlns="" class="EmphasisTypeItalic">B</em><sub>
                    <em xmlns="" class="EmphasisTypeItalic">j</em>
                  </sub><span class="stix">∈</span><em xmlns="" class="EmphasisTypeItalic">Blocks</em> <strong xmlns="" class="EmphasisTypeBold">do</strong></p>
                <p xmlns="http://www.w3.org/1999/xhtml" class="Para">4: <strong xmlns="" class="EmphasisTypeBold">if</strong> <em xmlns="" class="EmphasisTypeItalic">m</em>(<em xmlns="" class="EmphasisTypeItalic">Input</em>,<em xmlns="" class="EmphasisTypeItalic">P</em><sub>
                    <em xmlns="" class="EmphasisTypeItalic">in</em>
                  </sub>,<em xmlns="" class="EmphasisTypeItalic">B</em><sub>
                    <em xmlns="" class="EmphasisTypeItalic">j</em>
                  </sub>)≥<em xmlns="" class="EmphasisTypeItalic">Max</em> <strong xmlns="" class="EmphasisTypeBold">then</strong></p>
                <p xmlns="http://www.w3.org/1999/xhtml" class="Para">5: <em xmlns="" class="EmphasisTypeItalic">Max</em>←<em xmlns="" class="EmphasisTypeItalic">m</em>(<em xmlns="" class="EmphasisTypeItalic">Input</em>,<em xmlns="" class="EmphasisTypeItalic">P</em><sub>
                    <em xmlns="" class="EmphasisTypeItalic">in</em>
                  </sub>,<em xmlns="" class="EmphasisTypeItalic">B</em><sub>
                    <em xmlns="" class="EmphasisTypeItalic">j</em>
                  </sub>)</p>
                <p xmlns="http://www.w3.org/1999/xhtml" class="Para">6: <em xmlns="" class="EmphasisTypeItalic">B</em>←<em xmlns="" class="EmphasisTypeItalic">B</em><sub>
                    <em xmlns="" class="EmphasisTypeItalic">j</em>
                  </sub></p>
                <p xmlns="http://www.w3.org/1999/xhtml" class="Para">7: <strong xmlns="" class="EmphasisTypeBold">end if</strong></p>
                <p xmlns="http://www.w3.org/1999/xhtml" class="Para">8: <strong xmlns="" class="EmphasisTypeBold">end for</strong></p>
                <p xmlns="http://www.w3.org/1999/xhtml" class="Para">9: <strong xmlns="" class="EmphasisTypeBold">if</strong> <em xmlns="" class="EmphasisTypeItalic">Max</em>≤25<strong xmlns="" class="EmphasisTypeBold">then</strong></p>
                <p xmlns="http://www.w3.org/1999/xhtml" class="Para">10: <em xmlns="" class="EmphasisTypeItalic">Raws</em>←<em xmlns="" class="EmphasisTypeItalic">Input</em>(<em xmlns="" class="EmphasisTypeItalic">P</em><sub>
                    <em xmlns="" class="EmphasisTypeItalic">in</em>
                  </sub>,<em xmlns="" class="EmphasisTypeItalic">Max</em>)</p>
                <p xmlns="http://www.w3.org/1999/xhtml" class="Para">11: Add R(<em xmlns="" class="EmphasisTypeItalic">Raws</em>) to output</p>
                <p xmlns="http://www.w3.org/1999/xhtml" class="Para">12: <em xmlns="" class="EmphasisTypeItalic">P</em><sub>
                    <em xmlns="" class="EmphasisTypeItalic">in</em>
                  </sub>←<em xmlns="" class="EmphasisTypeItalic">P</em><sub>
                    <em xmlns="" class="EmphasisTypeItalic">in</em>
                  </sub> + <em xmlns="" class="EmphasisTypeItalic">Max</em></p>
                <p xmlns="http://www.w3.org/1999/xhtml" class="Para">13: <strong xmlns="" class="EmphasisTypeBold">else</strong></p>
                <p xmlns="http://www.w3.org/1999/xhtml" class="Para">14: <em xmlns="" class="EmphasisTypeItalic">P</em><sub>
                    <em xmlns="" class="EmphasisTypeItalic">raw</em>
                  </sub>← beginning of the longest match in <em xmlns="" class="EmphasisTypeItalic">B</em></p>
                <p xmlns="http://www.w3.org/1999/xhtml" class="Para">15: Add BC(<em xmlns="" class="EmphasisTypeItalic">B</em>) to output</p>
                <p xmlns="http://www.w3.org/1999/xhtml" class="Para">16: <strong xmlns="" class="EmphasisTypeBold">end if</strong></p>
                <p xmlns="http://www.w3.org/1999/xhtml" class="Para">17: <strong xmlns="" class="EmphasisTypeBold">end if</strong></p>
              </section><section id="Sec9" class="Section2 RenderAsSection2">
                <h3 class="Heading">Algorithm 4: ENCODE-REF Function</h3>
                <p xmlns="http://www.w3.org/1999/xhtml" class="Para">1: <em xmlns="" class="EmphasisTypeItalic">M</em>←0</p>
                <p xmlns="http://www.w3.org/1999/xhtml" class="Para">2: <strong xmlns="" class="EmphasisTypeBold">while</strong> <em xmlns="" class="EmphasisTypeItalic">Input</em>(<em xmlns="" class="EmphasisTypeItalic">P</em><sub>
                    <em xmlns="" class="EmphasisTypeItalic">in</em>
                  </sub>)=<em xmlns="" class="EmphasisTypeItalic">B</em>(<em xmlns="" class="EmphasisTypeItalic">P</em><sub>
                    <em xmlns="" class="EmphasisTypeItalic">raw</em>
                  </sub>)<strong xmlns="" class="EmphasisTypeBold">do</strong></p>
                <p xmlns="http://www.w3.org/1999/xhtml" class="Para">3: <em xmlns="" class="EmphasisTypeItalic">M</em>←<em xmlns="" class="EmphasisTypeItalic">M</em> + 1</p>
                <p xmlns="http://www.w3.org/1999/xhtml" class="Para">4: <em xmlns="" class="EmphasisTypeItalic">P</em><sub>
                    <em xmlns="" class="EmphasisTypeItalic">in</em>
                  </sub>←<em xmlns="" class="EmphasisTypeItalic">P</em><sub>
                    <em xmlns="" class="EmphasisTypeItalic">in</em>
                  </sub> + 1</p>
                <p xmlns="http://www.w3.org/1999/xhtml" class="Para">5: <em xmlns="" class="EmphasisTypeItalic">P</em><sub>
                    <em xmlns="" class="EmphasisTypeItalic">raw</em>
                  </sub>←<em xmlns="" class="EmphasisTypeItalic">P</em><sub>
                    <em xmlns="" class="EmphasisTypeItalic">raw</em>
                  </sub> + 1</p>
                <p xmlns="http://www.w3.org/1999/xhtml" class="Para">6: <strong xmlns="" class="EmphasisTypeBold">end while</strong></p>
                <p xmlns="http://www.w3.org/1999/xhtml" class="Para">7: Add RM(<em xmlns="" class="EmphasisTypeItalic">P</em><sub>
                    <em xmlns="" class="EmphasisTypeItalic">raw</em>
                  </sub>-M,<em xmlns="" class="EmphasisTypeItalic">M</em>) to output</p>
              </section><section id="Sec10" class="Section2 RenderAsSection2">
                <h3 class="Heading">Algorithm 5: ENCODE-RAW Function</h3>
                <p xmlns="http://www.w3.org/1999/xhtml" class="Para">1: <em xmlns="" class="EmphasisTypeItalic">Raws</em>←<em xmlns="" class="EmphasisTypeItalic">""</em></p>
                <p xmlns="http://www.w3.org/1999/xhtml" class="Para">2: <strong xmlns="" class="EmphasisTypeBold">while</strong> <em xmlns="" class="EmphasisTypeItalic">Input</em>(<em xmlns="" class="EmphasisTypeItalic">P</em><sub>
                    <em xmlns="" class="EmphasisTypeItalic">in</em>
                  </sub>)<span class="stix">∉</span>{<em xmlns="" class="EmphasisTypeItalic">A</em>,<em xmlns="" class="EmphasisTypeItalic">C</em>,<em xmlns="" class="EmphasisTypeItalic">G</em>,<em xmlns="" class="EmphasisTypeItalic">T</em>}<strong xmlns="" class="EmphasisTypeBold">do</strong></p>
                <p xmlns="http://www.w3.org/1999/xhtml" class="Para">3: <em xmlns="" class="EmphasisTypeItalic">Raws</em>←<em xmlns="" class="EmphasisTypeItalic">Raws</em><span class="stix">∘</span><em xmlns="" class="EmphasisTypeItalic">Input</em>[<em xmlns="" class="EmphasisTypeItalic">P</em><sub>
                    <em xmlns="" class="EmphasisTypeItalic">in</em>
                  </sub>];</p>
                <p xmlns="http://www.w3.org/1999/xhtml" class="Para">4: <em xmlns="" class="EmphasisTypeItalic">P</em><sub>
                    <em xmlns="" class="EmphasisTypeItalic">in</em>
                  </sub>←<em xmlns="" class="EmphasisTypeItalic">P</em><sub>
                    <em xmlns="" class="EmphasisTypeItalic">in</em>
                  </sub> + 1;</p>
                <p xmlns="http://www.w3.org/1999/xhtml" class="Para">5: <strong xmlns="" class="EmphasisTypeBold">end while</strong></p>
                <p xmlns="http://www.w3.org/1999/xhtml" class="Para">6: Add R(<em xmlns="" class="EmphasisTypeItalic">Raws</em>) to output</p>
                <p xmlns="http://www.w3.org/1999/xhtml" class="Para">In Algorithm 4 the number of matching characters between current input position and current reference position are determined and stored in variable <em xmlns="" class="EmphasisTypeItalic">M</em>. A reference entry RM(<em xmlns="" class="EmphasisTypeItalic">P</em><sub>
                    <em xmlns="" class="EmphasisTypeItalic">raw</em>
                  </sub>-M,<em xmlns="" class="EmphasisTypeItalic">M</em>) is added to the compressed output. The encoding of a raw sequence in Algorithm 5 is straight-forward: The string <em xmlns="" class="EmphasisTypeItalic">Raws</em> is filled with bases from the input until a <em xmlns="" class="EmphasisTypeItalic">normal</em> base is found and R(<em xmlns="" class="EmphasisTypeItalic">Raws</em>) is added to the output.</p>
                <p xmlns="http://www.w3.org/1999/xhtml" class="Para">Algorithm 3 requires more thoughts. The function FIND-MATCH is called, whenever there is a mismatch between the input and the reference. In this case, we need to find an alternative match for the current input position. First, we check whether there exists a match in the neighbourhood (mutations caused by few single nucleotide polymorphisms) of the current position of the reference block. The process is explained in detail later. If we cannot find a local match, then all the reference blocks are checked for a better match. The expression <em xmlns="" class="EmphasisTypeItalic">m</em>(<em xmlns="" class="EmphasisTypeItalic">Input</em>,<em xmlns="" class="EmphasisTypeItalic">P</em><sub>
                    <em xmlns="" class="EmphasisTypeItalic">in</em>
                  </sub>,<em xmlns="" class="EmphasisTypeItalic">B</em><sub>
                    <em xmlns="" class="EmphasisTypeItalic">j</em>
                  </sub>) returns the position of the longest prefix-suffix match of the current input in <em xmlns="" class="EmphasisTypeItalic">B</em><sub>
                    <em xmlns="" class="EmphasisTypeItalic">j</em>
                  </sub>.</p>
                <div xmlns="http://www.w3.org/1999/xhtml" class="Para">In our implementation of Algorithm 3, we care about the order in which all reference blocks are traversed. This is necessary, since loading blocks from the disk is a time consuming effort and should be avoided. Reference blocks are traversed with the following heuristic: <ol xmlns="" class="OrderedList"><li class="ListItem"><span xmlns="http://www.w3.org/1999/xhtml" class="ItemNumber">1.</span><div xmlns="http://www.w3.org/1999/xhtml" class="ItemContent">
                        <p class="Para">The block left and right of the current reference block</p>
                      </div><div class="ClearBoth">&nbsp;</div></li><li class="ListItem"><span xmlns="http://www.w3.org/1999/xhtml" class="ItemNumber">2.</span><div xmlns="http://www.w3.org/1999/xhtml" class="ItemContent">
                        <p class="Para">All other blocks on the same chromosome as the current reference block</p>
                      </div><div class="ClearBoth">&nbsp;</div></li><li class="ListItem"><span xmlns="http://www.w3.org/1999/xhtml" class="ItemNumber">3.</span><div xmlns="http://www.w3.org/1999/xhtml" class="ItemContent">
                        <p class="Para">All other blocks of the reference genome</p>
                      </div><div class="ClearBoth">&nbsp;</div></li></ol></div>
                <p xmlns="http://www.w3.org/1999/xhtml" class="Para">If there is no long enough match in any reference block, then we just encode few raw bases. Matches in other blocks are required to be longer than 25 characters, in order to avoid random matches. Our experiments have shown that the mere length of the human DNA causes a lot of unrelated matches with less than 20-25 characters.</p>
                <p xmlns="http://www.w3.org/1999/xhtml" class="Para">Algorithm 3 uses the function <em xmlns="" class="EmphasisTypeItalic">HasLocalAlternative</em>. The intuition is that we want to avoid searching all possible reference blocks all the time. Our experiments showed that one longest prefix-suffix-match lookup-operation in a single compressed suffix tree can take up to few milliseconds depending on the size. This is basically caused by the high bit-wise compression of the reference genome. Furthermore, potentially we have to load all the other reference block’s CTSs from the hard disk. Therefore, the naive strategy to search for a new reference block on <em xmlns="" class="EmphasisTypeItalic">each mismatch</em> does not scale.</p>
                <p xmlns="http://www.w3.org/1999/xhtml" class="Para">Instead we use a local search in the neighbourhood of the current input position and of the current reference position. This strategy has a biological foundation: Often two parts of a genome might only be different by few bases (base insertion, base removal, of base mutation). Whenever we find an appropriate match in the neighbourhood, we avoid checking other reference blocks, although they might contain better matches. In fact, if we searched for the best reference blocks each time, we could increase the compression rate slightly for the price of being orders of magnitude slower.</p>
                <p xmlns="http://www.w3.org/1999/xhtml" class="Para">The algorithm for local neighbourhood search is shown in Algorithm 6. If a match of at least 20 characters can be found near the current input and reference positions, then the difference until the match is encoded raw, and the match is encoded referentially.</p>
                <p xmlns="http://www.w3.org/1999/xhtml" class="Para">Decompression of the relative genome data is straightforward. Basically, all entries are processed according to their type. For decompression of genome data we do not need the compressed suffix trees any more, but only the raw blocks of the reference genome. The decompression of genome data is the least CPU-intensive task. Our evaluation in the next section shows that index generation and compression are CPU-bound, while the decompression phase is I/O-bound.</p>
              </section><section id="Sec11" class="Section2 RenderAsSection2">
                <h3 class="Heading">Algorithm 6: HAS-LOCAL-ALTERNATIVE Function</h3>
                <p xmlns="http://www.w3.org/1999/xhtml" class="Para">1: <strong xmlns="" class="EmphasisTypeBold">for</strong> <em xmlns="" class="EmphasisTypeItalic">i</em><span class="stix">∈</span>{0,1,…,6,7}<strong xmlns="" class="EmphasisTypeBold">do</strong></p>
                <p xmlns="http://www.w3.org/1999/xhtml" class="Para">2: <strong xmlns="" class="EmphasisTypeBold">for</strong> <em xmlns="" class="EmphasisTypeItalic">j</em><span class="stix">∈</span>{−7,−6,…,6,7}<strong xmlns="" class="EmphasisTypeBold">do</strong></p>
                <p xmlns="http://www.w3.org/1999/xhtml" class="Para">3: <strong xmlns="" class="EmphasisTypeBold">if</strong> <em xmlns="" class="EmphasisTypeItalic">Input</em>(<em xmlns="" class="EmphasisTypeItalic">P</em><sub>
                    <em xmlns="" class="EmphasisTypeItalic">in</em>
                  </sub> + <em xmlns="" class="EmphasisTypeItalic">i</em>,20)=<em xmlns="" class="EmphasisTypeItalic">B</em>(<em xmlns="" class="EmphasisTypeItalic">P</em><sub>
                    <em xmlns="" class="EmphasisTypeItalic">raw</em>
                  </sub> + <em xmlns="" class="EmphasisTypeItalic">j</em>,20)<strong xmlns="" class="EmphasisTypeBold">then</strong></p>
                <p xmlns="http://www.w3.org/1999/xhtml" class="Para">4: <strong xmlns="" class="EmphasisTypeBold">if</strong> <em xmlns="" class="EmphasisTypeItalic">i</em>≥1<strong xmlns="" class="EmphasisTypeBold">then</strong></p>
                <p xmlns="http://www.w3.org/1999/xhtml" class="Para">5: <em xmlns="" class="EmphasisTypeItalic">Raws</em>←<em xmlns="" class="EmphasisTypeItalic">Input</em>(<em xmlns="" class="EmphasisTypeItalic">P</em><sub>
                    <em xmlns="" class="EmphasisTypeItalic">in</em>
                  </sub>,<em xmlns="" class="EmphasisTypeItalic">i</em>)</p>
                <p xmlns="http://www.w3.org/1999/xhtml" class="Para">6: Add R(<em xmlns="" class="EmphasisTypeItalic">Raws</em>) to output</p>
                <p xmlns="http://www.w3.org/1999/xhtml" class="Para">7: <em xmlns="" class="EmphasisTypeItalic">P</em><sub>
                    <em xmlns="" class="EmphasisTypeItalic">in</em>
                  </sub>←<em xmlns="" class="EmphasisTypeItalic">P</em><sub>
                    <em xmlns="" class="EmphasisTypeItalic">in</em>
                  </sub> + <em xmlns="" class="EmphasisTypeItalic">i</em></p>
                <p xmlns="http://www.w3.org/1999/xhtml" class="Para">8: <strong xmlns="" class="EmphasisTypeBold">end if</strong></p>
                <p xmlns="http://www.w3.org/1999/xhtml" class="Para">9: <em xmlns="" class="EmphasisTypeItalic">P</em><sub>
                    <em xmlns="" class="EmphasisTypeItalic">raw</em>
                  </sub>←<em xmlns="" class="EmphasisTypeItalic">P</em><sub>
                    <em xmlns="" class="EmphasisTypeItalic">raw</em>
                  </sub> + <em xmlns="" class="EmphasisTypeItalic">j</em></p>
                <p xmlns="http://www.w3.org/1999/xhtml" class="Para">10: ENCODE-REF</p>
                <p xmlns="http://www.w3.org/1999/xhtml" class="Para">11: RETURN</p>
                <p xmlns="http://www.w3.org/1999/xhtml" class="Para">12: <strong xmlns="" class="EmphasisTypeBold">end if</strong></p>
                <p xmlns="http://www.w3.org/1999/xhtml" class="Para">13: <strong xmlns="" class="EmphasisTypeBold">end for</strong></p>
                <p xmlns="http://www.w3.org/1999/xhtml" class="Para">14: <strong xmlns="" class="EmphasisTypeBold">end for</strong></p>
                <p xmlns="http://www.w3.org/1999/xhtml" class="Para">We emphasize that our compression scheme is a heuristic, which mainly works when compressing a sequence with respect to a reference from the same species. The efficiency of this heuristic is evaluated in Section “Evaluation”. The worst-case time complexity of our compression algorithm is <em xmlns="" class="EmphasisTypeItalic">O</em>(—<em xmlns="" class="EmphasisTypeItalic">Input</em>—), since we need to traverse the whole input sequence one time from left to right and for each character we perform (in the worst case) one lookup in the reference. The complexity of the lookup depends on the length of the substring being looked up (we assume a fixed maximal match length) and is therefore constant. The worst-case space complexity is <em xmlns="" class="EmphasisTypeItalic">O</em>(—<em xmlns="" class="EmphasisTypeItalic">Input</em>—). Each character of the input is stored at most one time: either inside a raw block or as part of a referential block.</p>
              </section></div></section>
            <section xmlns="" xmlns:fn="http://www.w3.org/2005/xpath-functions" xmlns:meta="http://www.springer.com/app/meta" id="Sec12" class="Section1 RenderAsSection1"><h2 class="Heading js-ToggleCollapseSection">Related Work</h2><div class="js-CollapseSection"><p xmlns="http://www.w3.org/1999/xhtml" class="Para">In the following, we review existing work on biological data compression. In general, compression algorithms are either substitutional, statistical, or referential. Substitutional algorithms[<span class="CitationRef"><a href="#CR19">19</a></span>] replace long repeated substrings by references, e.g. Lempel-Ziv-based compression. Statistical algorithms[<span class="CitationRef"><a href="#CR13">13</a></span>, <span class="CitationRef"><a href="#CR20">20</a></span>] derive a predictive model from (a subset of) the input, based on partial matches. If the model always indicates high probabilities for the next symbol, then high compression rates are possible. While referential algorithms replace long substrings as well, the source for these subsets is usually not part of the input (the sequence to be compressed). Referential compression algorithms have drawn a lot of attention recently, since they allow for very high compression rates.</p><p xmlns="http://www.w3.org/1999/xhtml" class="Para">In[<span class="CitationRef"><a href="#CR16">16</a></span>], RLZ, a self-indexing based approach is proposed as follows: Given a self-index for a base sequence, compress other sequences with LZ77 encoding relative to the base sequence. In fact, a suffix array for the reference sequence is built and the reference entries are position references with length into the base sequence. Please note that the authors do not store raw sequences at any time, but only encode based on a list of references, no matter how long.</p><p xmlns="http://www.w3.org/1999/xhtml" class="Para">The authors of[<span class="CitationRef"><a href="#CR21">21</a></span>] proposed to store human genomes relatively to a reference chromosome. They used variable integers for storing absolute and relative positions of matches and represent often used <em xmlns="" class="EmphasisTypeItalic">k</em>-mers (sequences of length <em xmlns="" class="EmphasisTypeItalic">k</em>) with Huffman encoding.</p><p xmlns="http://www.w3.org/1999/xhtml" class="Para">In[<span class="CitationRef"><a href="#CR22">22</a></span>], a LZ77-style compression scheme is proposed. The main difference is that several reference sequences are taken into account. The match-finding process is based on hashing. Compression is performed on input blocks with shared Huffman models, in order to support random access. In[<span class="CitationRef"><a href="#CR23">23</a></span>], another LZ77-style compression scheme is proposed.</p><p xmlns="http://www.w3.org/1999/xhtml" class="Para">There exists further early work on non-referential compression algorithms[<span class="CitationRef"><a href="#CR24">24</a></span>–<span class="CitationRef"><a href="#CR27">27</a></span>]. Furthermore, there is compression based on context-free grammars[<span class="CitationRef"><a href="#CR28">28</a></span>], determination of Markov-model based probabilities of sequence positions[<span class="CitationRef"><a href="#CR29">29</a></span>], and hybrid methods[<span class="CitationRef"><a href="#CR30">30</a></span>]. In[<span class="CitationRef"><a href="#CR12">12</a></span>], a splay tree-based compression algorithm is proposed, which favors encoding of recently seen symbols. Finally, there exists previous work on relative compression of reads and alignment data, e.g.[<span class="CitationRef"><a href="#CR31">31</a></span>, <span class="CitationRef"><a href="#CR32">32</a></span>].</p></div></section>
            <section xmlns="" xmlns:fn="http://www.w3.org/2005/xpath-functions" xmlns:meta="http://www.springer.com/app/meta" id="Sec13" class="Section1 RenderAsSection1"><h2 class="Heading js-ToggleCollapseSection">Evaluation</h2><div class="js-CollapseSection"><p xmlns="http://www.w3.org/1999/xhtml" class="Para">In the following section, we evaluate our proposed compression scheme. All experiments have been run on a Acer Aspire 5950G with 16 GB RAM and Intel Core i7-2670QM, on Fedora 16 (64-Bit, Linux kernel 3.1). The code was implemented in C++, using the BOOST library[<span class="CitationRef"><a href="#CR33">33</a></span>], CST[<span class="CitationRef"><a href="#CR34">34</a></span>], and libz. All size measures are in byte, e.g. 1 MB means 1,000,000 bytes. The source code is available for download<sup>1</sup>.</p><p xmlns="http://www.w3.org/1999/xhtml" class="Para">First, we performed compression tests on human genomes. Our set of data genomes consists of 1000 genomes from the 1000 Genome project[<span class="CitationRef"><a href="#CR35">35</a></span>]. The 1000 Genome project group provides all sequenced genomes in Variant Call Format (VCF)[<span class="CitationRef"><a href="#CR36">36</a></span>] for download<sup>2</sup>. The Variant Call Format describes differences of sets of genomes with respect to a reference sequence, based on SNPs and indels. We have extracted one consensus sequence each for in total 1000 genomes. Since the project contains slightly more than 1000 genomes, we have only extracted the first 1000 genomes (columns from left to right) named in these VCF files. The consensus sequences for each chromosome of each genome were stored GZip-compressed on a hard disk. These 1000 GZip-compressed genomes need in total 700 GB of storage.</p><section id="Sec14" class="Section2 RenderAsSection2">
                <h3 class="Heading">Compression Ratio</h3>
                <div xmlns="http://www.w3.org/1999/xhtml" class="Para">First, we have compressed each chromosome referentially against the reference chromosome taken from HG19[<span class="CitationRef"><a href="#CR37">37</a></span>]. The results are shown in Figure<span class="InternalRef"><a href="#Fig2">2</a></span>. The first chromosome of 1000 humans needs 236.6 GB of uncompressed storage, while the referentially compressed file needs only 0.57 GB, yielding a compression ratio of 415:1. The smallest human chromosome of our 1000 genomes, Chromosome 22, is compressed from 36.4 GB down to 0.1 GB, yielding a compression ratio of 364:1. The overall compression ratio obtained for the 1000 human genomes is 397:1.<figure xmlns="" class="Figure" id="Fig2"><div xmlns="http://www.w3.org/1999/xhtml" class="MediaObject">
                      <img src="https://static-content.springer.com/image/art%3A10.1186%2F1748-7188-7-30/MediaObjects/13015_2012_Article_165_Fig2_HTML.jpg" alt="https://static-content.springer.com/image/art%3A10.1186%2F1748-7188-7-30/MediaObjects/13015_2012_Article_165_Fig2_HTML.jpg">
                    </div><figcaption class="Caption" lang="en"><div xmlns="http://www.w3.org/1999/xhtml" class="CaptionContent"><span class="CaptionNumber">Figure 2</span>
                        <p xmlns="" class="SimplePara"><strong class="EmphasisTypeBold">Compressed file size for different input chromosomes(in MB).</strong> We have compressed each chromosome referentially against the reference chromosome taken from HG19[<span xmlns="http://www.w3.org/1999/xhtml" class="CitationRef"><a href="#CR37">37</a></span>]. The results are shown in Figure<span xmlns="http://www.w3.org/1999/xhtml" class="InternalRef"><a href="#Fig2">2</a></span>. The overall compression ratio obtained for the 1000 human genomes is 397:1.</p>
                      </div></figcaption></figure></div>
                <p xmlns="http://www.w3.org/1999/xhtml" class="Para">For very similar sequences we achieve higher compression rates, than for less related sequences. This is inherent to all referential compression schemes: the more similar input and reference sequence are, longer referential matches can be found.</p>
                <div xmlns="http://www.w3.org/1999/xhtml" class="Para">We have evaluated the impact of the block size on the compression ratio. The results (average over all human genomes and chromosomes ) are shown in Figure<span class="InternalRef"><a href="#Fig3">3</a></span>. In general, a larger block size will yield better compression ratios. The intuition is that a larger block in main memory allows for finding longer matches. However, for a block size of 1 MB, our compression scheme can still achieve a compression ratio of 361:1, which is roughly competitive to existing relative compression schemes for human genome sequences. RLZ obtains a compression ration of 80:1 and RLZopt a compression ratio of 133:1 for human genomes. GDC achieves compression ratios of 200:1 - 500:1, depending on speed-tradeoffs. There exists one variant GDC-ultra, which achieves a compression ratio of 1000:1, which switches the reference sequence during compression. Switching the reference naturally allows for higher compression ratios. However, since we only use one reference, it seems to be fair to only compare our results to the non-ultra variant of GDC, obtaining a compression ratio of at most 500:1.<figure xmlns="" class="Figure" id="Fig3"><div xmlns="http://www.w3.org/1999/xhtml" class="MediaObject">
                      <img src="https://static-content.springer.com/image/art%3A10.1186%2F1748-7188-7-30/MediaObjects/13015_2012_Article_165_Fig3_HTML.jpg" alt="https://static-content.springer.com/image/art%3A10.1186%2F1748-7188-7-30/MediaObjects/13015_2012_Article_165_Fig3_HTML.jpg">
                    </div><figcaption class="Caption" lang="en"><div xmlns="http://www.w3.org/1999/xhtml" class="CaptionContent"><span class="CaptionNumber">Figure 3</span>
                        <p xmlns="" class="SimplePara"><strong class="EmphasisTypeBold">Compression ratio by block size.</strong> We have evaluated the impact of the block size on the compression ratio. The results (average over all human genomes and chromosomes ) are shown in Figure<span xmlns="http://www.w3.org/1999/xhtml" class="InternalRef"><a href="#Fig3">3</a></span>. In general, a larger block size will yield better compression ratios.</p>
                      </div></figcaption></figure></div>
                <p xmlns="http://www.w3.org/1999/xhtml" class="Para">Please note that the compression ratio for 300 MB is actually smaller than for 50MB. One explanation could be that all these matches found with smaller blocks allow for a shorter encoding than the matches found in a longer block. At first sight this might sound counter intuitive. To the best of our knowledge, no research has been conducted in this area, since the compression gain mainly depends on the choice of referential entries, i.e. how to encode positions, length entries and raw base entries. Referential compression is an optimization problem, where the longest matches often, but not necessarily, yield the shortest compressed representation. For instance, sometimes more (shorter) matches can be encoded more efficiently than less (longer) matches. We think that these effects are important to be studied in Future Work.</p>
                <p xmlns="http://www.w3.org/1999/xhtml" class="Para">The overall index size for the reference sequence (per chromosome) is in average 202 MB, while an average uncompressed input sequence is roughly 130 MB long. The size of the index structure is decreasing with decreasing block size, since the maximum length of paths in the suffix tree is limited.</p>
                <p xmlns="http://www.w3.org/1999/xhtml" class="Para">Additional experiments have been conducted on yeast genomes. We have downloaded<sup>3</sup> 39 yeast genomes, have chosen an arbitrary reference sequence (273614N) and referentially compressed the other 38 genomes (all chromosomes concatenated to each other) with respect to the reference. The average compression ratio is 61:1. The lower compression ratio compared to human sequences is not surprising, since it is well known that two yeast genomes can be less similar than even a human genome and a chimpanzee genome. In these cases (unoptimized) referential compression schemes do not obtain such nice compression ratios as with human genomes. RLZopt obtains compression ratios of 50:1, and GDC obtains compression ratios of 70:1-100:1, depending on speed-tradeoffs.</p>
                <p xmlns="http://www.w3.org/1999/xhtml" class="Para">In average, the index size of an average yeast gnome (around 12 MB) was found to be around 17.5 MB.</p>
              </section><section id="Sec15" class="Section2 RenderAsSection2">
                <h3 class="Heading">Compression Times</h3>
                <div xmlns="http://www.w3.org/1999/xhtml" class="Para">For serial compression, the compression algorithm performs compression on one block at a time (in a single thread). The memory requirements during compression are roughly 6*BS (one reference raw block, one input genome block, and the compressed suffix tree of the input block). In addition we use a write buffer for output operations (size: 1 MB), which can be neglected. The results for serial compression of the 1000 human genomes are shown in Figure<span class="InternalRef"><a href="#Fig4">4</a></span>. All the results were averaged over all chromosomes as input.<figure xmlns="" class="Figure" id="Fig4"><div xmlns="http://www.w3.org/1999/xhtml" class="MediaObject">
                      <img src="https://static-content.springer.com/image/art%3A10.1186%2F1748-7188-7-30/MediaObjects/13015_2012_Article_165_Fig4_HTML.jpg" alt="https://static-content.springer.com/image/art%3A10.1186%2F1748-7188-7-30/MediaObjects/13015_2012_Article_165_Fig4_HTML.jpg">
                    </div><figcaption class="Caption" lang="en"><div xmlns="http://www.w3.org/1999/xhtml" class="CaptionContent"><span class="CaptionNumber">Figure 4</span>
                        <p xmlns="" class="SimplePara"><strong class="EmphasisTypeBold">Compression ratio by block size.</strong> The results for serial compression of the 1000 human genomes are shown in Figure<span xmlns="http://www.w3.org/1999/xhtml" class="InternalRef"><a href="#Fig4">4</a></span>. All the results were averaged over all chromosomes as input. Clearly compression time decreases with increasing block index size, since less time is spent on traversing other blocks to find best matches.</p>
                      </div></figcaption></figure></div>
                <p xmlns="http://www.w3.org/1999/xhtml" class="Para">Clearly compression time decreases with increasing block index size, since less time is spent on traversing other blocks to find best matches. If the block index size is 1 MB (equals to roughly 6 MB main memory usage), the compression times are roughly 9 seconds per chromosome, i.e. around 200 seconds to compress a whole human genome. This yields a compression speed of around 15 MB/s. With a block index size of 300 MB (roughly 1.8 GB main memory usage), the compression time is down to 1.5 seconds per chromosome, i.e. around 35 seconds for compression a whole human genome. This yields a compression speed of 85 MB/s. The time necessary to create the compressed suffix tree for the reference is roughly 20 minutes. If one takes the position that indexing time should belong to the online search time, then the compression speed is reduced to 2.42 MB/s for a single genomes and to 82.87 MB/s for 1000 genomes. However, we are convinced that index construction should be taken as an offline task.</p>
                <p xmlns="http://www.w3.org/1999/xhtml" class="Para">RLZopt obtains a compression speed of 1.34 MB/s, while GDC achieves compression speeds of 4-35 MB/s, depending on speed-tradeoffs. Therefore, our adaptive algorithm seems to be competitive with existing relative compression schemes, while using less memory in a controllable (by changing the block size) way.</p>
                <p xmlns="http://www.w3.org/1999/xhtml" class="Para">In addition, we have performed experiments with 39 yeast genomes as in the previous subsection. The compression time is around 70 seconds for one yeast genome, yielding a compression speed of 0.17 MB/s. As expected, our local search heuristic does not work as good as with human genomes. Most of the time is spent on consulting the compressed suffix tree which often only yields very short matches of length 10-20. RLZopt has a compression speed of 1.6 MB/s and GDC achieves compression speeds of 2-33 MB/s, depending on speed-tradeoffs. In this case, it seems to be advantageous to use a hash table as an index structure instead of a suffix tree, as proposed by the authors of GDC.</p>
                <p xmlns="http://www.w3.org/1999/xhtml" class="Para">The parallelization on multi cores of our approach is straightforward. Block-processing can be easily distributed on several CPUs (or even machines). However, it has to be kept in mind that each compressor might work on different parts of the reference genome, which means that for <em xmlns="" class="EmphasisTypeItalic">n</em> compressor threads the memory usage is increased by the factor <em xmlns="" class="EmphasisTypeItalic">n</em>. The upper limit main memory usage for seven threads is 7 GB (maximum size of the complete index) + 7<span class="stix">∗</span><em xmlns="" class="EmphasisTypeItalic">BS</em>, if all index structures are loaded into main memory.</p>
                <p xmlns="http://www.w3.org/1999/xhtml" class="Para">Further investigation showed that the compression time in the latter case is dominated by loading the index files into main memory. Basically, in the beginning each thread requests a set of index blocks, which (due to parallel access) might be loaded in a partial random access manner.</p>
                <p xmlns="http://www.w3.org/1999/xhtml" class="Para">To counter act this effect, we have pre-bulk loaded all index structures at once after a reboot (taking roughly 80 seconds) and then performed compression. In this setting, a whole human genome can be compressed within 11 seconds with 7 threads.</p>
                <p xmlns="http://www.w3.org/1999/xhtml" class="Para">We think that this bulk-load scenario is actually quite realistic, as scientists often want to compress/decompress sets of genomes prior/after analysis.</p>
              </section><section id="Sec16" class="Section2 RenderAsSection2">
                <h3 class="Heading">Decompression Statistics</h3>
                <p xmlns="http://www.w3.org/1999/xhtml" class="Para">For all tests - independent of the index block size - we obtained an average decompression time of 22 seconds. Further parallelization did not improve these values, since the decompression algorithm is I/O-bound. During the compression around 3.1 GB for the decompressed genome are written on the hard disk. Given that we measured the write speed of the hard disk with 150 MB/s, it seems hard to further optimize the decompression time. We have also tried to write only compressed FASTA files on the disk. Then, however, the decompression takes several minutes, because the (self-referential) compression is CPU-bound. The decompression times for competitors are similar, i.e. RLZopt achieves 130 MB/s and GDC up to 150 MB/s. We are convinced that for most decompressor implementations the actual limitation is the hard disk write speed.</p>
              </section></div></section>
            <section xmlns="" xmlns:fn="http://www.w3.org/2005/xpath-functions" xmlns:meta="http://www.springer.com/app/meta" id="Sec17" class="Section1 RenderAsSection1"><h2 class="Heading js-ToggleCollapseSection">Conclusions</h2><div class="js-CollapseSection"><p xmlns="http://www.w3.org/1999/xhtml" class="Para">We have proposed an adaptive referential compression schemes for genomes. The compression speed can be controlled by varying the amount of main memory for the compressor. Our variant with lowest main memory footprint (12 MB) achieves similar compression rates and compression speeds, while using almost two orders of magnitude less main memory. Our greedy variant using 9 GB of main memory is 2-3 times as fast as the best known variant. Compression speed can be further improved by massive parallelization on different machines.</p><p xmlns="http://www.w3.org/1999/xhtml" class="Para">Further work should be done on improving the compression ratio. For instance, it would be possible to find an efficient encoding for inverse complement matches or approximate matches.</p><p xmlns="http://www.w3.org/1999/xhtml" class="Para">Investigations on inter-species referential compression is challenging. So far, referential compression only works well, if input and reference belong to the same species. The development of a multi-species reference sequence would allow for multi purpose genome compression algorithms. Our initial experiments with human genome compression with respect to a mouse genome indicate that the matches are usually very short and the advantages of referential compression are mitigated.</p></div></section>
            <section xmlns="" xmlns:fn="http://www.w3.org/2005/xpath-functions" xmlns:meta="http://www.springer.com/app/meta" id="Sec18" class="Section1 RenderAsSection1"><h2 class="Heading js-ToggleCollapseSection">Endnotes</h2><div class="js-CollapseSection"><p xmlns="http://www.w3.org/1999/xhtml" class="Para">
                <sup>1</sup>
                <span class="ExternalRef"><a href="http://www2.informatik.hu-berlin.de/~wandelt/blockcompression"><span class="RefSource">http://www2.informatik.hu-berlin.de/<span class="stix">∼</span>wandelt/blockcompression</span></a></span>
                <sup>2</sup>
                <span class="ExternalRef"><a href="ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/release/20110521/"><span class="RefSource">ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/release/20110521/</span></a></span>
                <sup>3</sup>
                <span class="ExternalRef"><a href="ftp://ftp.sanger.ac.uk/pub/dmc/yeast/latest"><span class="RefSource">ftp://ftp.sanger.ac.uk/pub/dmc/yeast/latest</span></a></span>
              </p></div></section>
          <section xmlns="" xmlns:fn="http://www.w3.org/2005/xpath-functions" xmlns:meta="http://www.springer.com/app/meta" class="Section1 RenderAsSection1" id="Declarations"><h2 class="Heading js-ToggleCollapseSection">
              Declarations
            </h2><div class="js-CollapseSection"><aside class="Appendix" id="App1">
              
                <h3 class="Heading">Authors' original submitted files for images</h3><div id="widgetContainer"></div>
                <div xmlns="http://www.w3.org/1999/xhtml" class="Para">Below are the links to the authors’ original submitted files for images.<div class="MediaObject" id="MOESM1">
                    <div xmlns="" class="esm-item "><div xmlns="http://www.w3.org/1999/xhtml" class="caption-container"><a class="filename" href="https://static-content.springer.com/esm/art%3A10.1186%2F1748-7188-7-30/MediaObjects/13015_2012_165_MOESM1_ESM.png">13015_2012_165_MOESM1_ESM.png</a>
                          <span xmlns="" class="SimplePara">Authors’ original file for figure 1</span>
                        </div></div>
                  </div><div class="MediaObject" id="MOESM2">
                    <div xmlns="" class="esm-item "><div xmlns="http://www.w3.org/1999/xhtml" class="caption-container"><a class="filename" href="https://static-content.springer.com/esm/art%3A10.1186%2F1748-7188-7-30/MediaObjects/13015_2012_165_MOESM2_ESM.png">13015_2012_165_MOESM2_ESM.png</a>
                          <span xmlns="" class="SimplePara">Authors’ original file for figure 2</span>
                        </div></div>
                  </div><div class="MediaObject" id="MOESM3">
                    <div xmlns="" class="esm-item "><div xmlns="http://www.w3.org/1999/xhtml" class="caption-container"><a class="filename" href="https://static-content.springer.com/esm/art%3A10.1186%2F1748-7188-7-30/MediaObjects/13015_2012_165_MOESM3_ESM.png">13015_2012_165_MOESM3_ESM.png</a>
                          <span xmlns="" class="SimplePara">Authors’ original file for figure 3</span>
                        </div></div>
                  </div><div class="MediaObject" id="MOESM4">
                    <div xmlns="" class="esm-item "><div xmlns="http://www.w3.org/1999/xhtml" class="caption-container"><a class="filename" href="https://static-content.springer.com/esm/art%3A10.1186%2F1748-7188-7-30/MediaObjects/13015_2012_165_MOESM4_ESM.png">13015_2012_165_MOESM4_ESM.png</a>
                          <span xmlns="" class="SimplePara">Authors’ original file for figure 4</span>
                        </div></div>
                  </div></div>
              
            </aside><aside xmlns="http://www.w3.org/1999/xhtml" class="ArticleNote ArticleNoteMisc"><h3 xmlns="" class="Heading">Competing interests</h3><p xmlns="" class="SimplePara">The authors declare that they have no competing interests.</p></aside><aside xmlns="http://www.w3.org/1999/xhtml" class="ArticleNote ArticleNoteMisc"><h3 xmlns="" class="Heading">Authors’ contributions</h3><p xmlns="" class="SimplePara">All authors have contributed equally to the main text. Sebastian Wandelt has implemented the presented algorithms on C++. All authors read and approved the final manuscript.</p></aside></div></section></div>
                            <div class="FulltextWrapper"><section xmlns="" xmlns:fn="http://www.w3.org/2005/xpath-functions" xmlns:meta="http://www.springer.com/app/meta" class="Section1 RenderAsSection1 Affiliations" id="Aff"><h2 class="Heading js-ToggleCollapseSection">Authors’ Affiliations</h2><div class="js-CollapseSection"><div class="Affiliation" id="Aff1"><span class="AffiliationNumber">(1)</span><div class="AffiliationText">Institute for Computer Science, Humboldt-Universität zu Berlin</div></div></div></section></div>
                            <div class="FulltextWrapper"><section xmlns="" xmlns:fn="http://www.w3.org/2005/xpath-functions" xmlns:meta="http://www.springer.com/app/meta" class="Section1 RenderAsSection1 SectionTypeMaterialsAndMethods" id="Bib1"><h2 class="Heading js-ToggleCollapseSection">References</h2><div class="js-CollapseSection"><ol class="BibliographyWrapper"><li class="Citation"><cite class="CitationContent" id="CR1">Initial sequencing and analysis of the human genome. Nature. 2001, 409 (6822): 860-921. 10.1038/35057062<span class="Occurrences"><span class="Occurrence OccurrenceDOI"><a target="_blank" href="http://dx.doi.org/10.1038/35057062"><span><span>View Article</span></span></a></span><span class="Occurrence OccurrenceGS"><a target="_blank" href="http://scholar.google.com/scholar_lookup?title=Initial%20sequencing%20and%20analysis%20of%20the%20human%20genome&amp;journal=Nature&amp;volume=409&amp;issue=6822&amp;pages=860-921&amp;publication_year=2001&amp;doi=10.1038%2F35057062"><span><span>Google Scholar</span></span></a></span></span></cite></li><li class="Citation"><cite class="CitationContent" id="CR2">Schadt EE, Turner S, Kasarskis A: A window into third-generation sequencing. Human Mol Genet. 2010, 19 (R2): R227-R240. 10.1093/hmg/ddq416. [<span xmlns="http://www.w3.org/1999/xhtml" class="ExternalRef"><a href="http://dx.doi.org/10.1093/hmg/ddq416"><span class="RefSource">http://dx.doi.org/10.1093/hmg/ddq416</span></a></span>]. 10.1093/hmg/ddq416<span class="Occurrences"><span class="Occurrence OccurrenceDOI"><a target="_blank" href="http://dx.doi.org/10.1093/hmg/ddq416"><span><span>View Article</span></span></a></span><span class="Occurrence OccurrenceGS"><a target="_blank" href="http://scholar.google.com/scholar_lookup?title=A%20window%20into%20third-generation%20sequencing&amp;author=EE.%20Schadt&amp;author=S.%20Turner&amp;author=A.%20Kasarskis&amp;journal=Human%20Mol%20Genet&amp;volume=19&amp;issue=R2&amp;pages=R227-R240&amp;publication_year=2010&amp;doi=10.1093%2Fhmg%2Fddq416"><span><span>Google Scholar</span></span></a></span></span></cite></li><li class="Citation"><cite class="CitationContent" id="CR3">International Cancer Genome Consortium Data Portal–a one-stop shop for cancer genomics data. Database : the journal of biological databases and curation. 2011, 2011 (0): bar026-[<span xmlns="http://www.w3.org/1999/xhtml" class="ExternalRef"><a href="http://dx.doi.org/10.1093/database/bar026"><span class="RefSource">http://dx.doi.org/10.1093/database/bar026</span></a></span>].<span class="Occurrences"></span></cite></li><li class="Citation"><cite class="CitationContent" id="CR4">Kahn SD: On the future of genomic data. Science. 2011, 331 (6018): 728-729. [<span xmlns="http://www.w3.org/1999/xhtml" class="ExternalRef"><a href="http://www.sciencemag.org/content/331/6018/728.abstract"><span class="RefSource">http://www.sciencemag.org/content/331/6018/728.abstract</span></a></span>]. 10.1126/science.1197891<span class="Occurrences"><span class="Occurrence OccurrencePID"><a target="_blank" href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=21311016"><span><span>PubMed</span></span></a></span><span class="Occurrence OccurrenceDOI"><a target="_blank" href="http://dx.doi.org/10.1126/science.1197891"><span><span>View Article</span></span></a></span><span class="Occurrence OccurrenceGS"><a target="_blank" href="http://scholar.google.com/scholar_lookup?title=On%20the%20future%20of%20genomic%20data&amp;author=SD.%20Kahn&amp;journal=Science&amp;volume=331&amp;issue=6018&amp;pages=728-729&amp;publication_year=2011&amp;doi=10.1126%2Fscience.1197891"><span><span>Google Scholar</span></span></a></span></span></cite></li><li class="Citation"><cite class="CitationContent" id="CR5">Fusaro VA, Patil P, Gafni E, Wall DP, Tonellato PJ: Biomedical cloud computing with amazon web services. PLoS Comput Biol. 2011, 7 (8): 1-6.[<span xmlns="http://www.w3.org/1999/xhtml" class="ExternalRef"><a href="https://sremote.pitt.edu:11018/login.aspx?direct=truedb=aphAN=67016557site=ehost-live"><span class="RefSource">https://sremote.pitt.edu:11018/login.aspx?direct=truedb=aphAN=67016557site=ehost-live</span></a></span>].<span class="Occurrences"><span class="Occurrence OccurrenceDOI"><a target="_blank" href="http://dx.doi.org/10.1371/journal.pcbi.1002147"><span><span>View Article</span></span></a></span><span class="Occurrence OccurrenceGS"><a target="_blank" href="http://scholar.google.com/scholar_lookup?title=Biomedical%20cloud%20computing%20with%20amazon%20web%20services&amp;author=VA.%20Fusaro&amp;author=P.%20Patil&amp;author=E.%20Gafni&amp;author=DP.%20Wall&amp;author=PJ.%20Tonellato&amp;journal=PLoS%20Comput%20Biol&amp;volume=7&amp;issue=8&amp;pages=1-6&amp;publication_year=2011"><span><span>Google Scholar</span></span></a></span></span></cite></li><li class="Citation"><cite class="CitationContent" id="CR6">Cloud computing and the DNA data race. Nat Biotechnol. 2010, 28 (7): 691-693. [<span xmlns="http://www.w3.org/1999/xhtml" class="ExternalRef"><a href="http://dx.doi.org/10.1038/nbt0710-691"><span class="RefSource">http://dx.doi.org/10.1038/nbt0710-691</span></a></span>].<span class="Occurrences"><span class="Occurrence OccurrenceDOI"><a target="_blank" href="http://dx.doi.org/10.1038/nbt0710-691"><span><span>View Article</span></span></a></span><span class="Occurrence OccurrenceGS"><a target="_blank" href="http://scholar.google.com/scholar_lookup?title=Cloud%20computing%20and%20the%20DNA%20data%20race&amp;author=VA.%20Fusaro&amp;author=P.%20Patil&amp;author=E.%20Gafni&amp;author=DP.%20Wall&amp;author=PJ.%20Tonellato&amp;journal=Nat%20Biotechnol&amp;volume=28&amp;issue=7&amp;pages=691-693&amp;publication_year=2010"><span><span>Google Scholar</span></span></a></span></span></cite></li><li class="Citation"><cite class="CitationContent" id="CR7">The case for cloud computing in genome informatics. Genome Biol. 2010, 11 (5): 207+-[<span xmlns="http://www.w3.org/1999/xhtml" class="ExternalRef"><a href="http://dx.doi.org/10.1186/gb-2010-11-5-207"><span class="RefSource">http://dx.doi.org/10.1186/gb-2010-11-5-207</span></a></span>].<span class="Occurrences"><span class="Occurrence OccurrenceDOI"><a target="_blank" href="http://dx.doi.org/10.1186/gb-2010-11-5-207"><span><span>View Article</span></span></a></span><span class="Occurrence OccurrenceGS"><a target="_blank" href="http://scholar.google.com/scholar_lookup?title=The%20case%20for%20cloud%20computing%20in%20genome%20informatics&amp;author=VA.%20Fusaro&amp;author=P.%20Patil&amp;author=E.%20Gafni&amp;author=DP.%20Wall&amp;author=PJ.%20Tonellato&amp;journal=Genome%20Biol&amp;volume=11&amp;issue=5&amp;pages=207&amp;publication_year=2010"><span><span>Google Scholar</span></span></a></span></span></cite></li><li class="Citation"><cite class="CitationContent" id="CR8">Trelles O, Prins P, Snir M, Jansen RC: Big data, but are we ready?. Nat Rev Genet. 2011, 12 (3): 224-[<span xmlns="http://www.w3.org/1999/xhtml" class="ExternalRef"><a href="http://dx.doi.org/10.1038/nrg2857-c1"><span class="RefSource">http://dx.doi.org/10.1038/nrg2857-c1</span></a></span>].<span class="Occurrences"><span class="Occurrence OccurrencePID"><a target="_blank" href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=21301471"><span><span>PubMed</span></span></a></span><span class="Occurrence OccurrenceDOI"><a target="_blank" href="http://dx.doi.org/10.1038/nrg2857-c1"><span><span>View Article</span></span></a></span><span class="Occurrence OccurrenceGS"><a target="_blank" href="http://scholar.google.com/scholar_lookup?title=Big%20data%2C%20but%20are%20we%20ready%3F&amp;author=O.%20Trelles&amp;author=P.%20Prins&amp;author=M.%20Snir&amp;author=RC.%20Jansen&amp;journal=Nat%20Rev%20Genet&amp;volume=12&amp;issue=3&amp;pages=224&amp;publication_year=2011"><span><span>Google Scholar</span></span></a></span></span></cite></li><li class="Citation"><cite class="CitationContent" id="CR9">Pennisim E: Will computers crash genomics?. Science. 2011, 331 (6018): 666-668. [<span xmlns="http://www.w3.org/1999/xhtml" class="ExternalRef"><a href="http://dx.doi.org/10.1126/science.331.6018.666"><span class="RefSource">http://dx.doi.org/10.1126/science.331.6018.666</span></a></span>]. 10.1126/science.331.6018.666<span class="Occurrences"><span class="Occurrence OccurrenceDOI"><a target="_blank" href="http://dx.doi.org/10.1126/science.331.6018.666"><span><span>View Article</span></span></a></span><span class="Occurrence OccurrenceGS"><a target="_blank" href="http://scholar.google.com/scholar_lookup?title=Will%20computers%20crash%20genomics%3F&amp;author=E.%20Pennisim&amp;journal=Science&amp;volume=331&amp;issue=6018&amp;pages=666-668&amp;publication_year=2011&amp;doi=10.1126%2Fscience.331.6018.666"><span><span>Google Scholar</span></span></a></span></span></cite></li><li class="Citation"><cite class="CitationContent" id="CR10">Giancarlo R, Scaturro D, Utro F: Textual data compression in computational biology: algorithmic techniques. Comput Sci Rev. January 2012, 6 (1): 1-25. 10.1016/j.cosrev.2011.11.001<span class="Occurrences"><span class="Occurrence OccurrenceDOI"><a target="_blank" href="http://dx.doi.org/10.1016/j.cosrev.2011.11.001"><span><span>View Article</span></span></a></span><span class="Occurrence OccurrenceGS"><a target="_blank" href="http://scholar.google.com/scholar_lookup?title=Textual%20data%20compression%20in%20computational%20biology%3A%20algorithmic%20techniques&amp;author=R.%20Giancarlo&amp;author=D.%20Scaturro&amp;author=F.%20Utro&amp;journal=Comput%20Sci%20Rev&amp;volume=6&amp;issue=1&amp;pages=1-25&amp;publication_year=2012&amp;doi=10.1016%2Fj.cosrev.2011.11.001"><span><span>Google Scholar</span></span></a></span></span></cite></li><li class="Citation"><cite class="CitationContent" id="CR11">Nalbantoglu ÖU, Russell DJ, Sayood K: Data compression concepts and algorithms and their applications to bioinformatics. Entropy. 2010, 12: 34-52. [<span xmlns="http://www.w3.org/1999/xhtml" class="ExternalRef"><a href="http://www.mdpi.com/1099-4300/12/1/34/"><span class="RefSource">http://www.mdpi.com/1099-4300/12/1/34/</span></a></span>].<span class="Occurrences"><span class="Occurrence OccurrencePID"><a target="_blank" href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=20157640"><span><span>PubMed</span></span></a></span><span class="Occurrence OccurrencePMCID"><a target="_blank" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2821113"><span><span>PubMed Central</span></span></a></span><span class="Occurrence OccurrenceDOI"><a target="_blank" href="http://dx.doi.org/10.3390/e12010034"><span><span>View Article</span></span></a></span><span class="Occurrence OccurrenceGS"><a target="_blank" href="http://scholar.google.com/scholar_lookup?title=Data%20compression%20concepts%20and%20algorithms%20and%20their%20applications%20to%20bioinformatics&amp;author=%C3%96U.%20Nalbantoglu&amp;author=DJ.%20Russell&amp;author=K.%20Sayood&amp;journal=Entropy&amp;volume=12&amp;pages=34-52&amp;publication_year=2010"><span><span>Google Scholar</span></span></a></span></span></cite></li><li class="Citation"><cite class="CitationContent" id="CR12">Antoniou D, Theodoridis E, Tsakalidis A: Compressing biological sequences using self adjusting data structures. 10th IEEE International Conference on Information Technology and Applications in Biomedicine. 2010, 1-5.<span class="Occurrences"><span class="Occurrence OccurrenceGS"><a target="_blank" href="http://scholar.google.com/scholar_lookup?title=Compressing%20biological%20sequences%20using%20self%20adjusting%20data%20structures&amp;author=D.%20Antoniou&amp;author=E.%20Theodoridis&amp;author=A.%20Tsakalidis&amp;pages=1-5&amp;publication_year=2010"><span><span>Google Scholar</span></span></a></span></span></cite></li><li class="Citation"><cite class="CitationContent" id="CR13">Pratas D, Pinho AJ: Compressing the Human Genome Using Exclusively Markov Models. PACBB, Volume 93 of Advances in Intelligent and Soft Computing. Edited by: Rocha MP, JMC Rodríguez, Fdez-Riverola F, Valencia A. 2011, 213-220. [<span xmlns="http://www.w3.org/1999/xhtml" class="ExternalRef"><a href="http://dblp.uni-trier.de/db/conf/pacbb/pacbb2011.html#PratasP11"><span class="RefSource">http://dblp.uni-trier.de/db/conf/pacbb/pacbb2011.html#PratasP11</span></a></span>].<span class="Occurrences"><span class="Occurrence OccurrenceGS"><a target="_blank" href="http://scholar.google.com/scholar_lookup?title=Compressing%20the%20Human%20Genome%20Using%20Exclusively%20Markov%20Models&amp;author=D.%20Pratas&amp;author=AJ.%20Pinho&amp;pages=213-220&amp;publication_year=2011"><span><span>Google Scholar</span></span></a></span></span></cite></li><li class="Citation"><cite class="CitationContent" id="CR14">Deorowicz S, Grabowski S: Robust relative compression of genomes with random access. Bioinformatics. 2011, 27 (21): 2979-2986. [<span xmlns="http://www.w3.org/1999/xhtml" class="ExternalRef"><a href="http://dx.doi.org/10.1093/bioinformatics/btr505"><span class="RefSource">http://dx.doi.org/10.1093/bioinformatics/btr505</span></a></span>]. 10.1093/bioinformatics/btr505<span class="Occurrences"><span class="Occurrence OccurrencePID"><a target="_blank" href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=21896510"><span><span>PubMed</span></span></a></span><span class="Occurrence OccurrenceDOI"><a target="_blank" href="http://dx.doi.org/10.1093/bioinformatics/btr505"><span><span>View Article</span></span></a></span><span class="Occurrence OccurrenceGS"><a target="_blank" href="http://scholar.google.com/scholar_lookup?title=Robust%20relative%20compression%20of%20genomes%20with%20random%20access&amp;author=S.%20Deorowicz&amp;author=S.%20Grabowski&amp;journal=Bioinformatics&amp;volume=27&amp;issue=21&amp;pages=2979-2986&amp;publication_year=2011&amp;doi=10.1093%2Fbioinformatics%2Fbtr505"><span><span>Google Scholar</span></span></a></span></span></cite></li><li class="Citation"><cite class="CitationContent" id="CR15">Kuruppu S, Puglisi S, Zobel J: Relative Lempel-Ziv compression of genomes for large-scale storage and retrieval. String Processing and Information Retrieval, Volume 6393 of Lecture Notes in Computer Science. Edited by: Chavez E, Lonardi S. Berlin / Heidelberg: Springer, 2010, 201-206.<span class="Occurrences"><span class="Occurrence OccurrenceGS"><a target="_blank" href="http://scholar.google.com/scholar_lookup?title=Relative%20Lempel-Ziv%20compression%20of%20genomes%20for%20large-scale%20storage%20and%20retrieval&amp;author=S.%20Kuruppu&amp;author=S.%20Puglisi&amp;author=J.%20Zobel&amp;pages=201-206&amp;publication_year=2010"><span><span>Google Scholar</span></span></a></span></span></cite></li><li class="Citation"><cite class="CitationContent" id="CR16">Kuruppu S, Puglisi SJ, Zobel J: Relative Lempel-Ziv compression of genomes for large-scale storage and retrieval. Proceedings of the 17th international conference on String processing and information retrieval, SPIRE’10. 2010, 201-206. [<span xmlns="http://www.w3.org/1999/xhtml" class="ExternalRef"><a href="http://dl.acm.org/citation.cfm?id=1928328.1928353"><span class="RefSource">http://dl.acm.org/citation.cfm?id=1928328.1928353</span></a></span>], Berlin, Heidelberg: Springer-Verlag.<span class="Occurrences"><span class="Occurrence OccurrenceDOI"><a target="_blank" href="http://dx.doi.org/10.1007/978-3-642-16321-0_20"><span><span>View Article</span></span></a></span><span class="Occurrence OccurrenceGS"><a target="_blank" href="http://scholar.google.com/scholar_lookup?title=Relative%20Lempel-Ziv%20compression%20of%20genomes%20for%20large-scale%20storage%20and%20retrieval&amp;author=S.%20Kuruppu&amp;author=SJ.%20Puglisi&amp;author=J.%20Zobel&amp;pages=201-206&amp;publication_year=2010"><span><span>Google Scholar</span></span></a></span></span></cite></li><li class="Citation"><cite class="CitationContent" id="CR17">Ukkonen E: On-line construction of suffix trees. Algorithmica. 1995, 14: 249-260. 10.1007/BF01206331. [<span xmlns="http://www.w3.org/1999/xhtml" class="ExternalRef"><a href="http://dx.doi.org/10.1007/BF01206331"><span class="RefSource">http://dx.doi.org/10.1007/BF01206331</span></a></span>]. 10.1007/BF01206331<span class="Occurrences"><span class="Occurrence OccurrenceDOI"><a target="_blank" href="http://dx.doi.org/10.1007/BF01206331"><span><span>View Article</span></span></a></span><span class="Occurrence OccurrenceGS"><a target="_blank" href="http://scholar.google.com/scholar_lookup?title=On-line%20construction%20of%20suffix%20trees&amp;author=E.%20Ukkonen&amp;journal=Algorithmica&amp;volume=14&amp;pages=249-260&amp;publication_year=1995&amp;doi=10.1007%2FBF01206331"><span><span>Google Scholar</span></span></a></span></span></cite></li><li class="Citation"><cite class="CitationContent" id="CR18">Ohlebusch E, Fischer J, Gog S: CST++. String Processing and Information Retrieval -17th International Symposium, SPIRE 2010. 2010, 322-333.<span class="Occurrences"><span class="Occurrence OccurrenceGS"><a target="_blank" href="http://scholar.google.com/scholar_lookup?title=CST%2B%2B&amp;author=E.%20Ohlebusch&amp;author=J.%20Fischer&amp;author=S.%20Gog&amp;pages=322-333&amp;publication_year=2010"><span><span>Google Scholar</span></span></a></span></span></cite></li><li class="Citation"><cite class="CitationContent" id="CR19">Kuruppu S, Beresford-Smith B, Conway T, Zobel J: Iterative dictionary construction for compression of large DNA data sets. IEEE/ACM Trans Comput Biol Bioinformatics. 2012, 9: 137-149. [<span xmlns="http://www.w3.org/1999/xhtml" class="ExternalRef"><a href="http://dx.doi.org/10.1109/TCBB.2011.82"><span class="RefSource">http://dx.doi.org/10.1109/TCBB.2011.82</span></a></span>].<span class="Occurrences"><span class="Occurrence OccurrenceDOI"><a target="_blank" href="http://dx.doi.org/10.1109/TCBB.2011.82"><span><span>View Article</span></span></a></span><span class="Occurrence OccurrenceGS"><a target="_blank" href="http://scholar.google.com/scholar_lookup?title=Iterative%20dictionary%20construction%20for%20compression%20of%20large%20DNA%20data%20sets&amp;author=S.%20Kuruppu&amp;author=B.%20Beresford-Smith&amp;author=T.%20Conway&amp;author=J.%20Zobel&amp;journal=IEEE%2FACM%20Trans%20Comput%20Biol%20Bioinformatics&amp;volume=9&amp;pages=137-149&amp;publication_year=2012"><span><span>Google Scholar</span></span></a></span></span></cite></li><li class="Citation"><cite class="CitationContent" id="CR20">Duc Cao M, Dix TI, Allison L, Mears C: A simple statistical algorithm for biological sequence compression. Proceedings of the 2007 Data Compression Conference. 2007, 43-52. [<span xmlns="http://www.w3.org/1999/xhtml" class="ExternalRef"><a href="http://dl.acm.org/citation.cfm?id=1251981.1252877"><span class="RefSource">http://dl.acm.org/citation.cfm?id=1251981.1252877</span></a></span>], IEEE Computer Society, Washington, DC, USA.<span class="Occurrences"><span class="Occurrence OccurrenceDOI"><a target="_blank" href="http://dx.doi.org/10.1109/DCC.2007.7"><span><span>View Article</span></span></a></span><span class="Occurrence OccurrenceGS"><a target="_blank" href="http://scholar.google.com/scholar_lookup?title=A%20simple%20statistical%20algorithm%20for%20biological%20sequence%20compression&amp;author=M.%20Duc%20Cao&amp;author=TI.%20Dix&amp;author=L.%20Allison&amp;author=C.%20Mears&amp;pages=43-52&amp;publication_year=2007"><span><span>Google Scholar</span></span></a></span></span></cite></li><li class="Citation"><cite class="CitationContent" id="CR21">Christley S, Lu Y, Li C, Xie X: Human genomes as email attachments. Bioinformatics. 2009, 25 (2): 274-275. [<span xmlns="http://www.w3.org/1999/xhtml" class="ExternalRef"><a href="http://dx.doi.org/10.1093/bioinformatics/btn582"><span class="RefSource">http://dx.doi.org/10.1093/bioinformatics/btn582</span></a></span>]. 10.1093/bioinformatics/btn582<span class="Occurrences"><span class="Occurrence OccurrencePID"><a target="_blank" href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=18996942"><span><span>PubMed</span></span></a></span><span class="Occurrence OccurrenceDOI"><a target="_blank" href="http://dx.doi.org/10.1093/bioinformatics/btn582"><span><span>View Article</span></span></a></span><span class="Occurrence OccurrenceGS"><a target="_blank" href="http://scholar.google.com/scholar_lookup?title=Human%20genomes%20as%20email%20attachments&amp;author=S.%20Christley&amp;author=Y.%20Lu&amp;author=C.%20Li&amp;author=X.%20Xie&amp;journal=Bioinformatics&amp;volume=25&amp;issue=2&amp;pages=274-275&amp;publication_year=2009&amp;doi=10.1093%2Fbioinformatics%2Fbtn582"><span><span>Google Scholar</span></span></a></span></span></cite></li><li class="Citation"><cite class="CitationContent" id="CR22">Grabowski S, Deorowicz S: Engineering relative compression of genomes. ArXiv. 2011, , [<span xmlns="http://www.w3.org/1999/xhtml" class="ExternalRef"><a href="http://arxiv.org/abs/1103.2351"><span class="RefSource">http://arxiv.org/abs/1103.2351</span></a></span>].<span class="Occurrences"><span class="Occurrence OccurrenceGS"><a target="_blank" href="http://scholar.google.com/scholar_lookup?title=Engineering%20relative%20compression%20of%20genomes&amp;author=S.%20Grabowski&amp;author=S.%20Deorowicz&amp;publication_year=2011%2C"><span><span>Google Scholar</span></span></a></span></span></cite></li><li class="Citation"><cite class="CitationContent" id="CR23">Kreft S, Navarro G: LZ77-like compression with fast random access. Proceedings of the 2010 Data Compression Conference, DCC ’10. 2010, 239-248. [<span xmlns="http://www.w3.org/1999/xhtml" class="ExternalRef"><a href="http://dx.doi.org/10.1109/DCC.2010.29"><span class="RefSource">http://dx.doi.org/10.1109/DCC.2010.29</span></a></span>], IEEE Computer Society, Washington, DC, USA.<span class="Occurrences"><span class="Occurrence OccurrenceDOI"><a target="_blank" href="http://dx.doi.org/10.1109/DCC.2010.29"><span><span>View Article</span></span></a></span><span class="Occurrence OccurrenceGS"><a target="_blank" href="http://scholar.google.com/scholar_lookup?title=LZ77-like%20compression%20with%20fast%20random%20access&amp;author=S.%20Kreft&amp;author=G.%20Navarro&amp;pages=239-248&amp;publication_year=2010"><span><span>Google Scholar</span></span></a></span></span></cite></li><li class="Citation"><cite class="CitationContent" id="CR24">Grumbach S, Tahi F: Compression of DNA sequences. Data Compression Conference. 1993, 340-350.<span class="Occurrences"><span class="Occurrence OccurrenceGS"><a target="_blank" href="http://scholar.google.com/scholar_lookup?title=Compression%20of%20DNA%20sequences&amp;author=S.%20Grumbach&amp;author=F.%20Tahi&amp;pages=340-350&amp;publication_year=1993"><span><span>Google Scholar</span></span></a></span></span></cite></li><li class="Citation"><cite class="CitationContent" id="CR25">Chen X, Kwong S, Li M: A compression algorithm for DNA sequences and its applications in genome comparison. Proceedings of the fourth annual international conference on Computational molecular biology, RECOMB ’00. 2000, 107-107. [<span xmlns="http://www.w3.org/1999/xhtml" class="ExternalRef"><a href="http://doi.acm.org/10.1145/332306.332352"><span class="RefSource">http://doi.acm.org/10.1145/332306.332352</span></a></span>], New York, NY, USA: ACM.<span class="Occurrences"><span class="Occurrence OccurrenceDOI"><a target="_blank" href="http://dx.doi.org/10.1145/332306.332352"><span><span>View Article</span></span></a></span><span class="Occurrence OccurrenceGS"><a target="_blank" href="http://scholar.google.com/scholar_lookup?title=A%20compression%20algorithm%20for%20DNA%20sequences%20and%20its%20applications%20in%20genome%20comparison&amp;author=X.%20Chen&amp;author=S.%20Kwong&amp;author=M.%20Li&amp;pages=107-107&amp;publication_year=2000"><span><span>Google Scholar</span></span></a></span></span></cite></li><li class="Citation"><cite class="CitationContent" id="CR26">Manzini G, Rastero M: A simple and fast DNA compressor. Software - Practice and Experience. 2004, 34: 1397-1411. 10.1002/spe.619<span class="Occurrences"><span class="Occurrence OccurrenceDOI"><a target="_blank" href="http://dx.doi.org/10.1002/spe.619"><span><span>View Article</span></span></a></span><span class="Occurrence OccurrenceGS"><a target="_blank" href="http://scholar.google.com/scholar_lookup?title=A%20simple%20and%20fast%20DNA%20compressor&amp;author=G.%20Manzini&amp;author=M.%20Rastero&amp;journal=Software%20-%20Practice%20and%20Experience&amp;volume=34&amp;pages=1397-1411&amp;publication_year=2004&amp;doi=10.1002%2Fspe.619"><span><span>Google Scholar</span></span></a></span></span></cite></li><li class="Citation"><cite class="CitationContent" id="CR27">Behzadi B, Le Fessant F: DNA compression challenge revisited: a dynamic programming approach. Combinatorial Pattern Matching, Volume 3537 of Lecture Notes in Computer Science. Edited by: Apostolico A, Crochemore M, Park K. Berlin / Heidelberg: Springer, 2005, 85-96.<span class="Occurrences"><span class="Occurrence OccurrenceGS"><a target="_blank" href="http://scholar.google.com/scholar_lookup?title=DNA%20compression%20challenge%20revisited%3A%20a%20dynamic%20programming%20approach&amp;author=B.%20Behzadi&amp;author=F.%20Le%20Fessant&amp;pages=85-96&amp;publication_year=2005"><span><span>Google Scholar</span></span></a></span></span></cite></li><li class="Citation"><cite class="CitationContent" id="CR28">Cherniavsky N, Ladner R: Grammar-based compression of DNA sequences. [Unpublished work], 2004.<span class="Occurrences"><span class="Occurrence OccurrenceGS"><a target="_blank" href="http://scholar.google.com/scholar_lookup?&amp;author=N.%20Cherniavsky&amp;author=R.%20Ladner&amp;publication_year=2004"><span><span>Google Scholar</span></span></a></span></span></cite></li><li class="Citation"><cite class="CitationContent" id="CR29">Cao MD, Dix TI, Allison L, Mears C: A simple statistical algorithm for biological sequence compression. Data Compression Conference. 2007, 0: 43-52.<span class="Occurrences"><span class="Occurrence OccurrenceGS"><a target="_blank" href="http://scholar.google.com/scholar_lookup?title=A%20simple%20statistical%20algorithm%20for%20biological%20sequence%20compression&amp;author=MD.%20Cao&amp;author=TI.%20Dix&amp;author=L.%20Allison&amp;author=C.%20Mears&amp;journal=Data%20Compression%20Conference&amp;volume=0&amp;pages=43-52&amp;publication_year=2007"><span><span>Google Scholar</span></span></a></span></span></cite></li><li class="Citation"><cite class="CitationContent" id="CR30">Matsumoto T, Sadakane K, Imai H: Biological sequence compression algorithms. Genome Informatics. 2000, 11: 43-52.<span class="Occurrences"><span class="Occurrence OccurrencePID"><a target="_blank" href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=11700586"><span><span>PubMed</span></span></a></span><span class="Occurrence OccurrenceGS"><a target="_blank" href="http://scholar.google.com/scholar_lookup?title=Biological%20sequence%20compression%20algorithms&amp;author=T.%20Matsumoto&amp;author=K.%20Sadakane&amp;author=H.%20Imai&amp;journal=Genome%20Informatics&amp;volume=11&amp;pages=43-52&amp;publication_year=2000"><span><span>Google Scholar</span></span></a></span></span></cite></li><li class="Citation"><cite class="CitationContent" id="CR31">Sakib MN, Tang J, Zheng WJ, Huang CT: Improving transmission efficiency of large sequence alignment/map (SAM) files. PLoS ONE. 2011, 6 (12): e28251-[<span xmlns="http://www.w3.org/1999/xhtml" class="ExternalRef"><a href="http://dx.doi.org/10.1371"><span class="RefSource">http://dx.doi.org/10.1371</span></a></span>]. 10.1371/journal.pone.0028251<span class="Occurrences"><span class="Occurrence OccurrencePID"><a target="_blank" href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=22164252"><span><span>PubMed</span></span></a></span><span class="Occurrence OccurrencePMCID"><a target="_blank" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3229529"><span><span>PubMed Central</span></span></a></span><span class="Occurrence OccurrenceDOI"><a target="_blank" href="http://dx.doi.org/10.1371/journal.pone.0028251"><span><span>View Article</span></span></a></span><span class="Occurrence OccurrenceGS"><a target="_blank" href="http://scholar.google.com/scholar_lookup?title=Improving%20transmission%20efficiency%20of%20large%20sequence%20alignment%2Fmap%20%28SAM%29%20files&amp;author=MN.%20Sakib&amp;author=J.%20Tang&amp;author=WJ.%20Zheng&amp;author=CT.%20Huang&amp;journal=PLoS%20ONE&amp;volume=6&amp;issue=12&amp;pages=e28251&amp;publication_year=2011&amp;doi=10.1371%2Fjournal.pone.0028251"><span><span>Google Scholar</span></span></a></span></span></cite></li><li class="Citation"><cite class="CitationContent" id="CR32">Hsi-Yang Fritz M, Leinonen R, Cochrane G, Birney E: Efficient storage of high throughput sequencing data using reference-based compression. Genome Res. 2011, 21 (5): 734-740. [<span xmlns="http://www.w3.org/1999/xhtml" class="ExternalRef"><a href="http://genome.cshlp.org/content/early/2011/01/18/gr.114819.110.abstract"><span class="RefSource">http://genome.cshlp.org/content/early/2011/01/18/gr.114819.110.abstract</span></a></span>]. 10.1101/gr.114819.110<span class="Occurrences"><span class="Occurrence OccurrencePID"><a target="_blank" href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=21245279"><span><span>PubMed</span></span></a></span><span class="Occurrence OccurrencePMCID"><a target="_blank" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3083090"><span><span>PubMed Central</span></span></a></span><span class="Occurrence OccurrenceDOI"><a target="_blank" href="http://dx.doi.org/10.1101/gr.114819.110"><span><span>View Article</span></span></a></span><span class="Occurrence OccurrenceGS"><a target="_blank" href="http://scholar.google.com/scholar_lookup?title=Efficient%20storage%20of%20high%20throughput%20sequencing%20data%20using%20reference-based%20compression&amp;author=M.%20Hsi-Yang%20Fritz&amp;author=R.%20Leinonen&amp;author=G.%20Cochrane&amp;author=E.%20Birney&amp;journal=Genome%20Res&amp;volume=21&amp;issue=5&amp;pages=734-740&amp;publication_year=2011&amp;doi=10.1101%2Fgr.114819.110"><span><span>Google Scholar</span></span></a></span></span></cite></li><li class="Citation"><cite class="CitationContent" id="CR33">BOOST C++ Libraries. [<span xmlns="http://www.w3.org/1999/xhtml" class="ExternalRef"><a href="http://www.boost.org"><span class="RefSource">http://www.boost.org</span></a></span>].]<span class="Occurrences"></span></cite></li><li class="Citation"><cite class="CitationContent" id="CR34">Ohlebusch E, Fischer J, Gog S: CST++. SPIRE’10. 2010, 322-333.<span class="Occurrences"><span class="Occurrence OccurrenceGS"><a target="_blank" href="http://scholar.google.com/scholar_lookup?title=CST%2B%2B&amp;author=E.%20Ohlebusch&amp;author=J.%20Fischer&amp;author=S.%20Gog&amp;pages=322-333&amp;publication_year=2010"><span><span>Google Scholar</span></span></a></span></span></cite></li><li class="Citation"><cite class="CitationContent" id="CR35">, : A map of human genome variation from population-scale sequencing. Nature. 2010, 467 (7319): 1061-1073. [<span xmlns="http://www.w3.org/1999/xhtml" class="ExternalRef"><a href="http://dx.doi.org/10.1038/nature09534"><span class="RefSource">http://dx.doi.org/10.1038/nature09534</span></a></span>]. 10.1038/nature09534<span class="Occurrences"><span class="Occurrence OccurrenceDOI"><a target="_blank" href="http://dx.doi.org/10.1038/nature09534"><span><span>View Article</span></span></a></span><span class="Occurrence OccurrenceGS"><a target="_blank" href="http://scholar.google.com/scholar_lookup?title=A%20map%20of%20human%20genome%20variation%20from%20population-scale%20sequencing&amp;journal=Nature&amp;volume=467&amp;issue=7319&amp;pages=1061-1073&amp;publication_year=2010&amp;doi=10.1038%2Fnature09534"><span><span>Google Scholar</span></span></a></span></span></cite></li><li class="Citation"><cite class="CitationContent" id="CR36">Danecek P, Auton A, Abecasis G, Albers CA, Banks E, DePristo MA, Handsaker RE, Lunter G, Marth GT, Sherry ST, McVean G, Durbin R, : The variant call format and VCFtools. Bioinformatics (Oxford, England). 2011, 27 (15): 2156-2158. 10.1093/bioinformatics/btr330. [<span xmlns="http://www.w3.org/1999/xhtml" class="ExternalRef"><a href="http://dx.doi.org/10.1093/bioinformatics/btr330"><span class="RefSource">http://dx.doi.org/10.1093/bioinformatics/btr330</span></a></span>]. 10.1093/bioinformatics/btr330<span class="Occurrences"><span class="Occurrence OccurrenceDOI"><a target="_blank" href="http://dx.doi.org/10.1093/bioinformatics/btr330"><span><span>View Article</span></span></a></span><span class="Occurrence OccurrenceGS"><a target="_blank" href="http://scholar.google.com/scholar_lookup?title=The%20variant%20call%20format%20and%20VCFtools&amp;author=P.%20Danecek&amp;author=A.%20Auton&amp;author=G.%20Abecasis&amp;author=CA.%20Albers&amp;author=E.%20Banks&amp;author=MA.%20DePristo&amp;author=RE.%20Handsaker&amp;author=G.%20Lunter&amp;author=GT.%20Marth&amp;author=ST.%20Sherry&amp;author=G.%20McVean&amp;author=R.%20Durbin&amp;journal=Bioinformatics%20%28Oxford%2C%20England%29&amp;volume=27&amp;issue=15&amp;pages=2156-2158&amp;publication_year=2011&amp;doi=10.1093%2Fbioinformatics%2Fbtr330"><span><span>Google Scholar</span></span></a></span></span></cite></li><li class="Citation"><cite class="CitationContent" id="CR37">Kent WJ, Sugnet CW, Furey TS, Roskin KM, Pringle TH, Zahler AM, Haussler D: The human genome browser at UCSC. Genome Res. 2002, 12 (6): 996-1006. [<span xmlns="http://www.w3.org/1999/xhtml" class="ExternalRef"><a href="http://www.ncbi.nlm.nih.gov/pubmed?cmd=Retrieve"><span class="RefSource">http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve#38;db=PubMed#38;dopt=Citation#38;list_uids=12045153</span></a></span>].<span class="Occurrences"><span class="Occurrence OccurrencePID"><a target="_blank" href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=12045153"><span><span>PubMed</span></span></a></span><span class="Occurrence OccurrencePMCID"><a target="_blank" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC186604"><span><span>PubMed Central</span></span></a></span><span class="Occurrence OccurrenceDOI"><a target="_blank" href="http://dx.doi.org/10.1101/gr.229102.%20Article%20published%20online%20before%20print%20in%20May%202002"><span><span>View Article</span></span></a></span><span class="Occurrence OccurrenceGS"><a target="_blank" href="http://scholar.google.com/scholar_lookup?title=The%20human%20genome%20browser%20at%20UCSC&amp;author=WJ.%20Kent&amp;author=CW.%20Sugnet&amp;author=TS.%20Furey&amp;author=KM.%20Roskin&amp;author=TH.%20Pringle&amp;author=AM.%20Zahler&amp;author=D.%20Haussler&amp;journal=Genome%20Res&amp;volume=12&amp;issue=6&amp;pages=996-1006&amp;publication_year=2002"><span><span>Google Scholar</span></span></a></span></span></cite></li></ol></div></section><section xmlns="" xmlns:fn="http://www.w3.org/2005/xpath-functions" xmlns:meta="http://www.springer.com/app/meta" class="Section1 RenderAsSection1" id="CopyrightMessage"><h2 class="Heading js-ToggleCollapseSection">Copyright</h2><div class="js-CollapseSection"><div xmlns="http://www.w3.org/1999/xhtml" class="ArticleCopyright">©&nbsp;Wandelt and Leser; licensee BioMed Central Ltd.&nbsp;2012<div class="CopyrightComment">
                <p xmlns="" class="SimplePara">This article is published under license to BioMed Central Ltd. This is an Open Access article distributed under the terms of the Creative Commons Attribution License (<span xmlns="http://www.w3.org/1999/xhtml" class="ExternalRef"><a href="http://creativecommons.org/licenses/by/2.0"><span class="RefSource">http://creativecommons.org/licenses/by/2.0</span></a></span>), which permits unrestricted use, distribution, and reproduction in any medium, provided the original work is properly cited.</p>
              </div></div></div></section></div>
                            


                        </div>
                    </div>
                    <div class="main__sidebar main__sidebar--right grid__col grid__col--xs-12 grid__col--md-3" id="main-sidebar">
                        
                        <div class="block">
                            <aside role="complementary">
                                <div class="SideBox">
                                    
    <p class="SideBox_action">
        <a class="SideBox_pdf" id="articlePdf" target="_blank" href="/track/pdf/10.1186/1748-7188-7-30?site&#x3D;almob.biomedcentral.com" data-track-event="click" data-event-category="PDF Download"
           data-event-action="Click download article PDF"
           data-event-label="10.1186/1748-7188-7-30">Download PDF</a>
    </p>

                                    
                                    
    <h3 class="export-citations-button" data-toggle="collapse" data-target="js-exportCitations">Export citations</h3>
    <div id="js-exportCitations" class="export-citations-content Panel Panel--s">
        <div class="Panel_section">
            <h4 class="Panel_sectionTitle">Citations & References</h4>
            <ul class="ListDownload" id="articleReferences">
                <li><a data-track-event="click" data-event-category="Article Page" data-event-action="Download Papers, Zotero, Reference Manager, RefWorks citations and references"
                       data-event-label="10.1186/1748-7188-7-30"
                       href="http://citation-needed.services.springer.com/v2/references/10.1186/1748-7188-7-30?format=refman&flavour=full">Papers, Zotero, Reference Manager, RefWorks <span>(.RIS)</span></a>
                </li>
                <li><a data-track-event="click" data-event-category="Article Page" data-event-action="Download EndNote citations and references" data-event-label="10.1186/1748-7188-7-30"
                       href="http://citation-needed.services.springer.com/v2/references/10.1186/1748-7188-7-30?format=endnote&flavour=full">EndNote <span>(.ENW)</span></a></li>
                <li><a data-track-event="click" data-event-category="Article Page" data-event-action="Download Mendeley, JabRef citations and references" data-event-label="10.1186/1748-7188-7-30"
                       href="http://citation-needed.services.springer.com/v2/references/10.1186/1748-7188-7-30?format=bibtex&flavour=full">Mendeley, JabRef <span>(.BIB)</span></a></li>
            </ul>
        </div>
        <div class="Panel_section">
            <h4 class="Panel_sectionTitle">Article citation</h4>
            <ul class="ListDownload">
                <li><a data-track-event="click" data-event-category="Article Page" data-event-action="Download Papers, Zotero, Reference Manager, RefWorks article citations" data-event-label="10.1186/1748-7188-7-30"
                       href="http://citation-needed.services.springer.com/v2/references/10.1186/1748-7188-7-30?format=refman&flavour=citation">Papers, Zotero, Reference Manager, RefWorks <span>(.RIS)</span></a></li>
                <li><a data-track-event="click" data-event-category="Article Page" data-event-action="Download EndNote article citations" data-event-label="10.1186/1748-7188-7-30"
                       href="http://citation-needed.services.springer.com/v2/references/10.1186/1748-7188-7-30?format=endnote&flavour=citation">EndNote <span>(.ENW)</span></a></li>
                <li><a data-track-event="click" data-event-category="Article Page" data-event-action="Download Mendeley, JabRef article citations" data-event-label="10.1186/1748-7188-7-30"
                       href="http://citation-needed.services.springer.com/v2/references/10.1186/1748-7188-7-30?format=bibtex&flavour=citation">Mendeley, JabRef <span>(.BIB)</span></a></li>
            </ul>
        </div>
        <div class="Panel_section">
            <h4 class="Panel_sectionTitle">References</h4>
            <ul class="ListDownload">
                <li><a data-track-event="click" data-event-category="Article Page" data-event-action="Download Papers, Zotero, Reference Manager, RefWorks references" data-event-label="10.1186/1748-7188-7-30"
                       href="http://citation-needed.services.springer.com/v2/references/10.1186/1748-7188-7-30?format=refman&flavour=references">Papers, Zotero, Reference Manager, RefWorks <span>(.RIS)</span></a>
                </li>
                <li><a data-track-event="click" data-event-category="Article Page" data-event-action="Download EndNote references" data-event-label="10.1186/1748-7188-7-30"
                       href="http://citation-needed.services.springer.com/v2/references/10.1186/1748-7188-7-30?format=endnote&flavour=references">EndNote <span>(.ENW)</span></a></li>
                <li><a data-track-event="click" data-event-category="Article Page" data-event-action="Download Mendeley, JabRef references" data-event-label="10.1186/1748-7188-7-30"
                       href="http://citation-needed.services.springer.com/v2/references/10.1186/1748-7188-7-30?format=bibtex&flavour=references">Mendeley, JabRef <span>(.BIB)</span></a></li>
            </ul>
        </div>
    </div>

                                </div>
                                <div class="SideBox">
                                    <nav xmlns="" xmlns:fn="http://www.w3.org/2005/xpath-functions" xmlns:meta="http://www.springer.com/app/meta" id="ArticleNavigation" class="table-of-contents" aria-label="table of contents"><div class="table-of-contents__content" data-sticky="toc"><h3 class="table-of-contents__title is-active" data-toggle="collapse" data-target="tocMenu" data-event-category="Table of Contents" data-track-event="click" data-event-action="Toggle ToC" data-event-label="Toggle">Table of Contents</h3><ul class="table-of-contents__menu is-open" role="menu" id="tocMenu"><li role="menuitem" class="table-of-contents__item"><a class="table-of-contents__link" href="#Abs1" data-event-category="Table of Contents" data-track-event="click" data-event-action="Link Clicked" data-event-label="Abstract">Abstract</a></li><li role="menuitem" class="table-of-contents__item"><a class="table-of-contents__link" href="#Sec1" data-event-category="Table of Contents" data-track-event="click" data-event-action="Link Clicked" data-event-label="Background">Background</a></li><li role="menuitem" class="table-of-contents__item"><a class="table-of-contents__link" href="#Sec2" data-event-category="Table of Contents" data-track-event="click" data-event-action="Link Clicked" data-event-label="General Idea">General Idea</a></li><li role="menuitem" class="table-of-contents__item"><a class="table-of-contents__link" href="#Sec4" data-event-category="Table of Contents" data-track-event="click" data-event-action="Link Clicked" data-event-label="Genome Compression">Genome Compression</a></li><li role="menuitem" class="table-of-contents__item"><a class="table-of-contents__link" href="#Sec12" data-event-category="Table of Contents" data-track-event="click" data-event-action="Link Clicked" data-event-label="Related Work">Related Work</a></li><li role="menuitem" class="table-of-contents__item"><a class="table-of-contents__link" href="#Sec13" data-event-category="Table of Contents" data-track-event="click" data-event-action="Link Clicked" data-event-label="Evaluation">Evaluation</a></li><li role="menuitem" class="table-of-contents__item"><a class="table-of-contents__link" href="#Sec17" data-event-category="Table of Contents" data-track-event="click" data-event-action="Link Clicked" data-event-label="Conclusions">Conclusions</a></li><li role="menuitem" class="table-of-contents__item"><a class="table-of-contents__link" href="#Sec18" data-event-category="Table of Contents" data-track-event="click" data-event-action="Link Clicked" data-event-label="Endnotes">Endnotes</a></li><li role="menuitem" class="table-of-contents__item"><a class="table-of-contents__link" href="#Declarations" data-event-category="Table of Contents" data-track-event="click" data-event-action="Link Clicked" data-event-label="Declarations">
                        Declarations
                      </a></li><li role="menuitem" class="table-of-contents__item"><a class="table-of-contents__link" href="#Bib1" data-event-category="Table of Contents" data-track-event="click" data-event-action="Link Clicked" data-event-label="References">References</a></li><li role="menuitem" class="table-of-contents__item is-hidden"><a class="table-of-contents__link" href="#comments" data-event-category="Table of Contents" data-track-event="click" data-event-action="Link Clicked" data-event-label="Comments">Comments</a></li></ul></div></nav>
                                </div>
                                <div class="SideBox">
                                    <h3 class="SideBox_title">Metrics</h3>
                                    <ul class="SideBox_list list-stacked list-stacked--small list-stacked--2">
                                        <div id="article-access-count-loader" class="placeholder" data-replace="true" data-placeholder="/placeholder/v1/article/access-count/10.1186/1748-7188-7-30 #accessCount-container"></div>
                                        <div id="citation-count-loader" class="placeholder" data-replace="true" data-placeholder="/placeholder/v1/article/citation-count/10.1186/1748-7188-7-30 #count-container"></div>
                                        <div id="altmetric-loader" class="placeholder" data-replace="true" data-placeholder="/placeholder/v1/article/altmetric/10.1186/1748-7188-7-30 #altmetric-container"></div>
                                    </ul>
                                </div>
                                
    <div id="sociaMediaShare" class="SideBox">
        <h3 class="SideBox_title">Share this article</h3>
        <ul id="shareButtons" class="share">
            <li class="share__item">
                <button id="shareOnTwitter" class="js-btnShareOnTwitter share__button share__button--twitter" data-track-event="click"
                        data-event-category="Article Page" data-event-action="Share Twitter" data-event-label="10.1186/1748-7188-7-30">
                    <span class="sr-only">Share on Twitter</span>
                </button>
            </li>
            <li class="share__item">
                <button id="shareOnFacebook" class="js-btnShareOnFacebook share__button share__button--facebook" data-track-event="click"
                        data-event-category="Article Page" data-event-action="Share Facebook" data-event-label="10.1186/1748-7188-7-30">
                    <span class="sr-only">Share on Facebook</span>
                </button>
            </li>
            <li class="share__item">
                <button id="shareOnLinkedIn" class="js-btnShareOnLinkedIn share__button share__button--linkedin" data-track-event="click"
                        data-event-category="Article Page" data-event-action="Share LinkedIn" data-event-label="10.1186/1748-7188-7-30">
                    <span class="sr-only">Share on LinkedIn</span>
                </button>
            </li>
            <li class="share__item">
                <button id="shareOnWeibo" class="js-btnShareOnWeibo share__button share__button--weibo" data-track-event="click"
                        data-event-category="Article Page" data-event-action="Share Weibo" data-event-label="10.1186/1748-7188-7-30">
                    <span class="sr-only">Share on Weibo</span>
                </button>
            </li>
            <li class="share__item">
                <button id="shareOnGooglePlus" class="js-btnShareOnGplus share__button share__button--gplus" data-track-event="click"
                        data-event-category="Article Page" data-event-action="Share Google+" data-event-label="10.1186/1748-7188-7-30">
                    <span class="sr-only">Share on Google Plus</span>
                </button>
            </li>
            <li class="share__item">
                <button id="shareOnReddit" class="js-btnShareOnReddit share__button share__button--reddit" data-track-event="click"
                        data-event-category="Article Page" data-event-action="Share Reddit" data-event-label="10.1186/1748-7188-7-30">
                    <span class="sr-only">Share on Reddit</span>
                </button>
            </li>
        </ul>
    </div>

                                
    

                                
    

                                
    

                                <div class="SideBox">
                                    <h3 class="SideBox_title">Other Actions</h3>
                                    <ul class="list-stacked list-stacked--2 list-stacked--small">
                                        <li class="list-stacked__item">
                                            
    <a data-track-event="click" data-event-category="Article Page" data-event-action="Order article reprints" data-event-label="10.1186/1748-7188-7-30" href="https://www.odysseypress.com/onlinehost/reprint_order.php?type=A&page=0&journal=738&doi=10.1186%2F1748-7188-7-30&volume=7&issue=1&title=Adaptive+efficient+compression+of+genomes&author_name=Sebastian+Wandelt&start_page=1&end_page=9">Order reprint</a>

                                        </li>
                                    </ul>
                                </div>
                                
        
    
                            </aside>
                            
    <div id="FulltextAd" class="adsbox Ad Ad--skyscraper">
        <div class="Ad_content">
            <p class="Ad_label">Advertisement</p>
            <span id="pbgrd-43415139"></span>
        </div>
    </div>

                        </div>
                    </div>
                </div>
            </div>
        </main>
        <p style="display: none;" class="footer-dynamic-legal">
    <a class="footer-dynamic-legal__link" title="View Legal Information" href="http://www.miibeian.gov.cn">
        &#27818;ICP&#22791;15051854&#21495;-3
    </a>
</p>
        
            <div class="journal-footer">
                <div class="journal-footer__inner">
                    <div class="journal-footer__summary">
                        <h4 class="journal-footer__title">
                            Algorithms for Molecular Biology
                        </h4>
                        <p class="journal-footer__issn">ISSN: 1748-7188</p>
                    </div>
                    
                        <div class="journal-footer__contact">
                            <h4 class="journal-footer__contact-title">Contact us</h4>
                            <ul class="journal-footer__contact-list">
                                
                                    <li class="journal-footer__contact-item">Editorial email: <a href="mailto:amb@biomedcentral.com">amb@biomedcentral.com</a></li>
                                
                                
                                    <li class="journal-footer__contact-item">Support email: <a href="mailto:info@biomedcentral.com">info@biomedcentral.com</a></li>
                                
                            </ul>
                        </div>
                    
                </div>
            </div>
        
         
            <footer>
                <div class="footer">
                    <div class="footer__header">
                        <img srcset="/static/images/biomed-central-logo-white.c238298a.svg" src="/static/images/biomed-central-logo-white.faac4202.png" class="logo__img footer__logo" alt="Biomed Central" />
                        
                            <h2 class="footer__menu-title">Publisher Main Menu</h2>
                            <ul class="footer__primary-menu">
                                
                                    <li class="footer__primary-item">
                                        <a class="footer__link footer__primary-link" href="//www.biomedcentral.com/journals">Explore journals</a>
                                    </li>
                                
                                    <li class="footer__primary-item">
                                        <a class="footer__link footer__primary-link" href="//www.biomedcentral.com/getpublished">Get published</a>
                                    </li>
                                
                                    <li class="footer__primary-item">
                                        <a class="footer__link footer__primary-link" href="//www.biomedcentral.com/about">About BioMed Central</a>
                                    </li>
                                
                            </ul>
                        
                    </div>
                    <div id="Test-FooterExists" class="footer__content">
                        <div class="footer__legal">
                            <p class="footer__terms">By continuing to use this website, you agree to our <a class="footer__link footer__link--inline" href="//www.biomedcentral.com/terms-and-conditions">Terms and Conditions</a>, <a class="footer__link footer__link--inline"
                                    href="//www.biomedcentral.com/terms-and-conditions/privacy-statement">Privacy
                                statement</a> and <a class="footer__link footer__link--inline" href="//www.biomedcentral.com/about/cookies">Cookies</a> policy.</p>
                        </div>
                        
    <nav id="publisherFooter" class="footer__nav">
        <h3 class="footer__menu-title">Publisher secondary menu</h3>
        <div class="footer__nav-inner">
            
            <div class="footer__menu">
                <ul class="footer__list">
                    
                    <li class="footer__item">
                        <a class="footer__link" href="//www.biomedcentral.com/about/contact-us">Contact us</a>
                        
                    </li>
                    
                    <li class="footer__item">
                        <a class="footer__link" href="//www.biomedcentral.com/about/jobs">Jobs</a>
                        
                    </li>
                    
                    <li class="footer__item">
                        <a class="footer__link" href="//www.biomedcentral.com/logon?url&#x3D;/my/manuscripts">Manage manuscripts</a>
                        
                    </li>
                    
                    <li class="footer__item">
                        <a class="footer__link" href="//www.biomedcentral.com/logon?url&#x3D;/my/preferences">Sign up for article alerts</a>
                        
                    </li>
                    
                    <li class="footer__item">
                        <a class="footer__link" href="//www.biomedcentral.com/logon?url&#x3D;/my/preferences">Manage article alerts</a>
                        
                    </li>
                    
                    <li class="footer__item">
                        <a class="footer__link" href="https://biomedcentral.typeform.com/to/ve8NOd">Leave feedback</a>
                        
                    </li>
                    
                </ul>
            </div>
            
            <div class="footer__menu">
                <ul class="footer__list">
                    
                    <li class="footer__item">
                        <a class="footer__link" href="//www.biomedcentral.com/about/press-centre">Press center</a>
                        
                    </li>
                    
                    <li class="footer__item">
                        <a class="footer__link" href="http://blogs.biomedcentral.com/">Read more on our blogs</a>
                        
                    </li>
                    
                </ul>
            </div>
            
            <div class="footer__menu">
                <ul class="footer__list">
                    
                    <li class="footer__item">
                        <a class="footer__link" href="//www.biomedcentral.com/about/policies">Policies</a>
                        
                    </li>
                    
                    <li class="footer__item">
                        <a class="footer__link" href="//www.biomedcentral.com/licensing">Licensing</a>
                        
                    </li>
                    
                    <li class="footer__item">
                        <a class="footer__link" href="//www.biomedcentral.com/terms-and-conditions">Terms and conditions</a>
                        
                    </li>
                    
                    <li class="footer__item">
                        <a class="footer__link" href="//www.biomedcentral.com/terms-and-conditions/privacy-statement">Privacy statement</a>
                        
                    </li>
                    
                    <li class="footer__item">
                        <a class="footer__link" href="//www.biomedcentral.com/accessibility">Accessibility</a>
                        
                    </li>
                    
                    <li class="footer__item">
                        <a class="footer__link" href="//www.biomedcentral.com/about/cookies">Cookies</a>
                        
                    </li>
                    
                </ul>
            </div>
            

            <div class="footer__menu footer__menu--follow">
                <div class="footer__follow">
                    <h3 class="footer__follow-title">Follow BioMed Central</h3>
                    <ul class="follow follow--inline">
                        
                        <li class="footer__follow-item follow__item">
                            <a class="follow__link follow__link--twitter" href="https://twitter.com/biomedcentral" target="_blank" data-track-event="click" data-event-category="Social Media" data-event-action="Click Follow Social Media" data-event-label="Twitter">
                                <span class="follow__label">Twitter</span>
                            </a>
                        </li>
                        
                        
                        <li class="footer__follow-item follow__item">
                            <a class="follow__link follow__link--facebook" href="https://www.facebook.com/BioMedCentral" target="_blank" data-track-event="click" data-event-category="Social Media" data-event-action="Click Follow Social Media" data-event-label="Facebook">
                                <span class="follow__label">Facebook</span>
                            </a>
                        </li>
                        
                        
                        <li class="footer__follow-item follow__item">
                            <a class="follow__link follow__link--gplus" href="https://plus.google.com/106226337984152901734/" target="_blank" data-track-event="click" data-event-category="Social Media" data-event-action="Click Follow Social Media"
                               data-event-label="Google Plus">
                                <span class="follow__label">Google Plus</span>
                            </a>
                        </li>
                        
                        
                        <li class="footer__follow-item follow__item">
                            <a class="follow__link follow__link--youtube" href="https://www.youtube.com/user/BioMedCentral" target="_blank" data-track-event="click" data-event-category="Social Media" data-event-action="Click Follow Social Media" data-event-label="YouTube">
                                <span class="follow__label">YouTube</span>
                            </a>
                        </li>
                        
                        
                        <li class="footer__follow-item follow__item">
                            <a class="follow__link follow__link--linkedin" href="https://www.linkedin.com/company/biomed-central" target="_blank" data-track-event="click" data-event-category="Social Media" data-event-action="Click Follow Social Media" data-event-label="LinkedIn">
                                <span class="follow__label">LinkedIn</span>
                            </a>
                        </li>
                        
                        
                        <li class="footer__follow-item follow__item">
                            <a class="follow__link follow__link--reddit" href="https://www.reddit.com/domain/biomedcentral.com" target="_blank" data-track-event="click" data-event-category="Social Media" data-event-action="Click Follow Social Media" data-event-label="Reddit">
                                <span class="follow__label">Reddit</span>
                            </a>
                        </li>
                        
                        
                        <li class="footer__follow-item follow__item">
                            <a class="follow__link follow__link--weibo" href="http://www.weibo.com/biomedcentral" target="_blank" data-track-event="click" data-event-category="Social Media" data-event-action="Click Follow Social Media" data-event-label="Weibo">
                                <span class="follow__label">Weibo</span>
                            </a>
                        </li>
                        
                    </ul>
                </div>
            </div>
        </div>
    </nav>


                    </div>
                </div>
                <div class="branding-bar">
                    <div class="logo branding-bar__logo">
                        <img srcset="/static/images/logo-springernature.06a51479.svg" src="/static/images/logo-springernature.7c95d137.png" class="logo__img branding-bar-logo__img" alt="Springer Nature Logo" />
                    </div>
                    <p class="branding-bar__copyright">&#169; 2016 BioMed Central Ltd unless otherwise stated. Part of <a class="branding-bar__link" href="http://www.springernature.com">Springer Nature</a>.</p>
                </div>
            </footer>
        
    </div>
    
    <img class='tracker' style='display:none' src='/track/article/10.1186/1748-7188-7-30' alt=""/>

    <div id="JFulltext"></div>

    
    <script type="text/javascript" src="//cdn.pbgrd.com/bmc.js" async="async"></script>

    

    
<script>
(function() {
    var linkEl = document.querySelector('.js-ctm');
    if (window.matchMedia && window.matchMedia(linkEl.media)) {
        var script = document.createElement('script');
        script.src = '/static/js/oscar.7836530d.js';
        script.async = true;
        document.body.appendChild(script);
    }
})();
</script>



<script type="text/javascript">
    if (config) {
        (function (d) {
            Array.prototype.forEach.call(d.getElementsByClassName('RealMediaAd'), function (adElement) {
                var iframe = d.createElement('iframe');
                iframe.frameBorder = '0';
                iframe.scrolling = 'no';
                var position = adElement.getAttribute("data-position"),
                        iframeInDoc = adElement.getElementsByClassName('Ad_content')[0].appendChild(iframe),
                        doc = iframeInDoc.contentWindow.document;

                doc.open().write('<body onload="window.location.href = '
                        + '\'/placeholder/v1/advertising/realMedia?siteWithPath=' + encodeURIComponent(config.siteWithPath)
                        + '&placement=' + position + '\';">');
                doc.close();
            });
        })(document);
    }
</script>



    
        <script type="text/javascript">
            document.addEventListener('readystatechange', function () {
                if (document.readyState === 'complete') {
                    setTimeout(function () {
                        var script = document.createElement("script");
                        script.type = "text/javascript";
                        script.src = "https://recommendations.springernature.com/latest/entry-point.js";
                        document.getElementsByTagName("head")[0].appendChild(script);
                    }, 0);
                }
            });
        </script>
    


    <script type="text/javascript">
        var _kiq = _kiq || [];
        (function () {
            setTimeout(function () {
                var d = document, f = d.getElementsByTagName('script')[0], s = d.createElement('script');
                s.type = 'text/javascript';
                s.async = true;
                s.src = '//s3.amazonaws.com/ki.js/47412/9CC.js';
                f.parentNode.insertBefore(s, f);
            }, 1);
            setTimeout(function () {
                var d = document, f = d.getElementsByTagName('script')[0], s = d.createElement('script');
                s.type = 'text/javascript';
                s.async = true;
                s.src = '//s3.amazonaws.com/ki.js/47412/aC6.js';
                f.parentNode.insertBefore(s, f);
            }, 1);
        })();
    </script>


<img hidden src="https://verify.nature.com/verify/nature.png" border="0" width="0" height="0" style="display: none">
</body>
</html>
